---
title: "Pv_MIP_analyses"
author: "ZPH"
date: "2023-04-04"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/")
```

To start with, you need to run some code in Terminal to exclude untargeted sites and output the headers so that you can concatenate the VCFs.When I tried doing this within R, it broke my computer so...just copy and paste to Terminal for now.

```{bash}
cd /mnt/c/Users/zpopkinh/OneDrive\ -\ University\ of\ North\ Carolina\ at\ Chapel\ Hill/Pvivax\ MIP\ Design/Analysis/NAMRU_repool/Combined_Analysis/

#mkdir Combined_Analysis

#apt-get install bcftools

bcftools view -T targets.vcf.gz -O z -o FST_targeted_variants.vcf.gz FST_wrangler/analysis/variants.vcf.gz

bcftools view -T targets.vcf.gz -O z -o MAF5_targeted_variants.vcf.gz MAF5_wrangler/analysis/variants.vcf.gz

bcftools view -T targets.vcf.gz -O z -o MAF40_targeted_variants.vcf.gz MAF40_wrangler/analysis/variants.vcf.gz

for i in *.vcf.gz; do bcftools view -h ${i} > ${i%_targeted_variants.vcf.gz}_header.txt; done
```

Before moving on, let's look at missingness so that we can exclude missing samples as well and only have to do things once.

```{r}
library(vcfR)
library(tibble)
library(ggplot2)
library(RColorBrewer)

setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

FST_vcf <- read.vcfR("FST_for_concat.vcf.gz")
FST_dp <- extract.gt(FST_vcf, element='DP', as.numeric=TRUE)
FST_NAs <- sum(is.na(FST_dp[,1]))
FST_sample_missingness <- apply(FST_dp, MARGIN = 2, function(x){ sum(is.na(x)) })
FST_sample_missingness <- FST_sample_missingness/nrow(FST_vcf)
palette(brewer.pal(n=12, name = 'Set3'))
par(mar = c(12,4,4,2))

barplot(FST_sample_missingness, las = 2, col = 1:12)
title(ylab = "Missingness (%)")

par(mar = c(5,4,4,2))
FST_variant_missingness <- apply(FST_dp, MARGIN = 1, function(x){ sum(is.na(x)) })
FST_variant_missingness <- FST_variant_missingness/ncol(FST_vcf@gt[,-1])
hist(FST_variant_missingness, col = "#8DD3C7", xlab = "Missingness (%)")

missing_FST_samples <- names(FST_sample_missingness[FST_sample_missingness > 0.5])

missing_FST_samples

missing_FST_variants <- FST_variant_missingness[FST_variant_missingness > 0.5]

missing_FST_variants

MAF5_vcf <- read.vcfR("MAF5_for_concat.vcf.gz")
MAF5_dp <- extract.gt(MAF5_vcf, element='DP', as.numeric=TRUE)
MAF5_NAs <- sum(is.na(MAF5_dp[,1]))
MAF5_sample_missingness <- apply(MAF5_dp, MARGIN = 2, function(x){ sum(is.na(x)) })
MAF5_sample_missingness <- MAF5_sample_missingness/nrow(MAF5_vcf)
palette(brewer.pal(n=12, name = 'Set3'))
par(mar = c(12,4,4,2))

barplot(MAF5_sample_missingness, las = 2, col = 1:12)
title(ylab = "Missingness (%)")

par(mar = c(5,4,4,2))
MAF5_variant_missingness <- apply(MAF5_dp, MARGIN = 1, function(x){ sum(is.na(x)) })
MAF5_variant_missingness <- MAF5_variant_missingness/ncol(MAF5_vcf@gt[,-1])
hist(MAF5_variant_missingness, col = "#8DD3C7", xlab = "Missingness (%)")

missing_MAF5_samples <- names(MAF5_sample_missingness[MAF5_sample_missingness > 0.5])

missing_MAF5_samples

missing_MAF5_variants <- MAF5_variant_missingness[MAF5_variant_missingness > 0.5]

missing_MAF5_variants

MAF40_vcf <- read.vcfR("MAF40_for_concat.vcf.gz")
MAF40_dp <- extract.gt(MAF40_vcf, element='DP', as.numeric=TRUE)
MAF40_NAs <- sum(is.na(MAF40_dp[,1]))
MAF40_sample_missingness <- apply(MAF40_dp, MARGIN = 2, function(x){ sum(is.na(x)) })
MAF40_sample_missingness <- MAF40_sample_missingness/nrow(MAF40_vcf)
palette(brewer.pal(n=12, name = 'Set3'))
par(mar = c(12,4,4,2))

barplot(MAF40_sample_missingness, las = 2, col = 1:12)
title(ylab = "Missingness (%)")

par(mar = c(5,4,4,2))
MAF40_variant_missingness <- apply(MAF40_dp, MARGIN = 1, function(x){ sum(is.na(x)) })
MAF40_variant_missingness <- MAF40_variant_missingness/ncol(MAF40_vcf@gt[,-1])
hist(MAF40_variant_missingness, col = "#8DD3C7", xlab = "Missingness (%)")

missing_MAF40_samples <- names(MAF40_sample_missingness[MAF40_sample_missingness > 0.5])

missing_MAF40_samples

missing_MAF40_variants <- MAF40_variant_missingness[MAF40_variant_missingness > 0.5]

missing_MAF40_variants

missing_SNP_samples <- c(missing_FST_samples, missing_MAF5_samples, missing_MAF40_samples) |> unique() 

data.table::fwrite(as.list(missing_SNP_samples), file = "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/missing_SNP_samples.txt", sep = "\n", row.names = F, col.names = F)

PvG_vcf <- read.vcfR("../PvG_wrangler/analysis/variants.vcf.gz")
PvG_dp <- extract.gt(PvG_vcf, element='DP', as.numeric=TRUE)
PvG_NAs <- sum(is.na(PvG_dp[,1]))
PvG_sample_missingness <- apply(PvG_dp, MARGIN = 2, function(x){ sum(is.na(x)) })
PvG_sample_missingness <- PvG_sample_missingness/nrow(PvG_vcf)
palette(brewer.pal(n=12, name = 'Set3'))
par(mar = c(12,4,4,2))

barplot(PvG_sample_missingness, las = 2, col = 1:12)
title(ylab = "Missingness (%)")

par(mar = c(5,4,4,2))
PvG_variant_missingness <- apply(PvG_dp, MARGIN = 1, function(x){ sum(is.na(x)) })
PvG_variant_missingness <- PvG_variant_missingness/ncol(PvG_vcf@gt[,-1])
hist(PvG_variant_missingness, col = "#8DD3C7", xlab = "Missingness (%)")

missing_PvG_samples <- names(PvG_sample_missingness[PvG_sample_missingness > 0.5])

missing_PvG_samples

missing_PvG_variants <- PvG_variant_missingness[PvG_variant_missingness > 0.5]

missing_PvG_variants 

data.table::fwrite(as.list(names(missing_PvG_variants)), file = "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/missing_PvG_variants.txt", sep = "\n", row.names = F, col.names = F)

remove(FST_vcf, MAF5_vcf, MAF40_vcf, PvG_vcf)
```

In Notepad++ or another text editor, open the headers, delete everything but the sample list, then use a regex replace to change \t to \n so you can easily read it in. We are going to find the "outersect" of samples to exclude in R.

```{r}
FST_header <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/FST_header.txt", header = F, data.table = F)

MAF5_header <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/MAF5_header.txt", header = F, data.table = F)

MAF40_header <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/MAF40_header.txt", header = F, data.table = F)

outersect <- function(x, y) {
       sort(c(setdiff(x, y),
                           setdiff(y, x)))
}

outersect1 <- outersect(FST_header$V1, MAF5_header$V1)
outersect2 <- outersect(FST_header$V1, MAF40_header$V1)
outersect3 <- outersect(MAF5_header$V1, MAF40_header$V1)

exclude <- c(outersect1, outersect2, outersect3) |> unique()

exclude2 <- FST_header |> dplyr::filter(stringr::str_detect(FST_header$V1, "(?i)pvg|(?i)ntp|(?i)ntc"))

exclude3 <- MAF5_header |> dplyr::filter(stringr::str_detect(V1, "(?i)pvg|(?i)ntp|(?i)ntc"))

exclude4 <- MAF40_header |> dplyr::filter(stringr::str_detect(V1, "(?i)pvg|(?i)ntp|(?i)ntc"))

exclude <- c(exclude, exclude2$V1, exclude3$V1, exclude4$V1) |> unique()

exclude <- c(exclude, missing_SNP_samples) |> unique()

data.table::fwrite(as.list(exclude), file = "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/exclude.txt", sep = "\n", row.names = F, col.names = F)
```

Back to bash!

```{bash}
for i in *targeted_variants.vcf.gz; do bcftools view -S ^exclude.txt --force-samples -O z -o ${i%_targeted_variants.vcf.gz}_for_concat.vcf.gz ${i}; done

bcftools query -l FST_for_concat.vcf.gz | sort > sorted_samples.txt

for i in *for_concat.vcf.gz; do bcftools view -S sorted_samples.txt -O z -o ${i%_for_concat.vcf.gz}_sorted.vcf.gz ${i}; bcftools index ${i%_for_concat.vcf.gz}_sorted.vcf.gz; done

bcftools concat -a -D FST_sorted.vcf.gz MAF5_sorted.vcf.gz MAF40_sorted.vcf.gz | bcftools annotate --rename-chrs chr_rename.txt -O z -o renamed_concatenated_SNPs.vcf.gz

bcftools index renamed_concatenated_SNPs.vcf.gz

bcftools view -m2 -M2 -O z -o biallelics.vcf.gz renamed_concatenated_SNPs.vcf.gz

bcftools index biallelics.vcf.gz

#To filter by SNP missingness:

vcftools --gzvcf biallelics.vcf.gz --max-missing 0.5 --out biallelics_maxmissing05 --recode --recode-INFO-all

#Not important at this stage, but you will need to do the below in order to use rehh, simplify COI calculations, etc.

bcftools view -i 'AC>0' renamed_concatenated_SNPs.vcf.gz -Oz -o SNPs_no_gaps.vcf.gz
```

Now we have all the files we need. Time to load libraries, data and metadata. We will also produce our initial PCA before we start doing more fine-tuned analyses.

Transitioning to the Bob Verity method, so use this chunk to experiment:

```{r}
library(MIPanalyzer)

library(vcfR)

library(dplyr)

library(tibble)

library(summarytools)

setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

SNP_VCF <- read.vcfR("biallelics_maxmissing05.recode.vcf")

SNP_MIPs <- vcf2mipanalyzer_biallelic(vcfR = SNP_VCF) |> filter_loci_invariant(description = "filter loci to drop invariant sites")
SNP_wsaf <- get_wsaf(SNP_MIPs, impute = TRUE, FUN = mean)

pca <- MIPanalyzer::pca_wsaf(as.matrix(SNP_wsaf))

plot_df <- data.frame(PC1 = pca$x[,1],
                      PC2 = pca$x[,2])

pca_metadata <- read.csv("../../NAMRU_full_run/Combined_Analysis/metadata.csv")
pca_metadata[pca_metadata == ''] <- "Unknown"

pca_metadata$City = ""

Alto_Nanay <- list("DIAMANTE AZUL")
for (var in Alto_Nanay) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Alto Nanay"
}

Belen <- list("PADRECOCHA", "PROGRESO")
for (var in Belen) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Belén"
}

Indiana <- list("SAN LUIS", "SANTA ROSA")
for (var in Indiana) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Indiana"
}

Iquitos <- list("FRAY MARTIN", "IQUITOS", "LAGUNAS", "MANACAMIRI", "SANTA RITA")
for (var in Iquitos) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Iquitos"
}

Mazan <- list("PUERTO ALEGRE")
for (var in Mazan) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Mazán"
}

Napo <- list("SANTA MARIA")
for (var in Napo) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Napo"
}

Nauta <- list("NAUTA")
for (var in Nauta) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Nauta"
}

Pardo_Miguel <- list("YARINAL")
for (var in Pardo_Miguel) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Pardo Miguel"
}

Pichanaqui <- list("MILAGRO")
for (var in Pichanaqui) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Pichanaqui"
}

Punchana <- list("CENTRO FUERTE", "FLOR DE AGOSTO", "MOMONCILLO", "NUEVA YORK", "NUEVO SAN ANTONIO", "PORVENIR", "PUERTO ALICIA", "SAN ANDRES", "SANTO TOMAS", "GEN GEN")
for (var in Punchana) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Punchana"
}

Ramon_Castilla <- list("CABALLOCOCHA", "VISTA ALEGRE")
for (var in Ramon_Castilla) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Ramón Castilla"
}

San_Juan_Bautista <- list("LLANCHAMA", "NINARUMI", "NUEVA VIDA", "PUERTO ALMENDRA", "QUISTOCOCHA", "SAN JUAN", "SANTA CLARA", "ZUNGAROCOCHA")
for (var in San_Juan_Bautista) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "San Juan Bautista"
}

Tigre <- list("NUEVO RETIRO")
for (var in Tigre) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Tigre"
}

Torres_Causana <- list("PUERTO ELVIRA")
for (var in Torres_Causana) {
  print(var)
  pca_metadata$City[grepl(var, pca_metadata$Community)] <- "Torres Causana"
}

pca_metadata$Region = ""

Chanchamayo <- list("Pichanaqui")
for (var in Chanchamayo) {
  print(var)
  pca_metadata$Region[grepl(var, pca_metadata$City)] <- "Chanchamayo"
}

Loreto <- list("Tigre")
for (var in Loreto) {
  print(var)
  pca_metadata$Region[grepl(var, pca_metadata$City)] <- "Loreto"
}

Mariscal_Ramon_Castilla <- list("Ramón Castilla")
for (var in Mariscal_Ramon_Castilla) {
  print(var)
  pca_metadata$Region[grepl(var, pca_metadata$City)] <- "Mariscal Ramón Castilla"
}

Maynas <- list("Alto Nanay", "Belén", "Indiana", "Iquitos", "Mazán", "Napo", "Nauta", "Punchana", "San Juan Bautista", "Torres Causana")
for (var in Maynas) {
  print(var)
  pca_metadata$Region[grepl(var, pca_metadata$City)] <- "Maynas"
}

Rioja <- list("Pardo Miguel")
for (var in Rioja) {
  print(var)
  pca_metadata$Region[grepl(var, pca_metadata$City)] <- "Rioja"
}

pca_metadata$State = ""

Junin <- list("Chanchamayo")
for (var in Junin) {
  print(var)
  pca_metadata$State[grepl(var, pca_metadata$Region)] <- "Junín"
}

Loreto <- list("Loreto", "Mariscal Ramón Castilla", "Maynas")
for (var in Loreto) {
  print(var)
  pca_metadata$State[grepl(var, pca_metadata$Region)] <- "Loreto"
}

San_Martin <- list("Rioja")
for (var in San_Martin) {
  print(var)
  pca_metadata$State[grepl(var, pca_metadata$Region)] <- "San Martín"
}

pca_metadata <- pca_metadata %>% dplyr::mutate(CityYear = paste0(City, Collection_Year))

plot_df <- plot_df |> rownames_to_column(var = "Sample")

plot_df$Sample <- plot_df$Sample |> stringr::str_remove(pattern = "-SNP")

pca_metadata$Sample <- paste0(pca_metadata$Sample, "-NAMRU-1")

by <- dplyr::join_by(Sample)

pca_metadata_filtered <- dplyr::semi_join(pca_metadata, plot_df, by)

pca_metadata_filtered[pca_metadata_filtered == ''] <- "Unknown"

pca_metadata_filtered[pca_metadata_filtered == 'NA'] <- "Unknown"

pca_metadata_filtered |> data.table::fwrite("filtered_metadata.csv")

pca_metadata_filtered |> write_rds("filtered_metadata.rds")

plot_df <- right_join(plot_df, pca_metadata_filtered) |> subset(Community != "Unknown") |> droplevels()

city_n <- freq(plot_df$City)

year_n <- freq(plot_df$Collection_Year)

cityyear_n <- freq(plot_df$CityYear)

view(city_n, file = "samples_by_city.html" )

view(year_n, file = "samples_by_year.html")

view(cityyear_n, file = "samples_by_city_year.html")
```
Let's make some plots!

```{r}
library(viridis)

library(ggplot2)


setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Community))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA Community.png", width = 15, height = 10, units = "in", dpi = 600)

ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Health_Center))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA Health Center.png", width = 15, height = 10, units = "in", dpi = 600)

ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA Collection Year.png", width = 15, height = 10, units = "in", dpi = 600)

ggplot(plot_df, aes(x = PC1, y = PC2, shape = City, color = factor(Collection_Year))) +
  geom_point(aes(group = interaction(City, Collection_Year))) +
  scale_shape_manual(values = 1:nlevels(as.factor(plot_df$City))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA CityYear.png", device = "png", width = 15, height = 12, units = "in", dpi = 600)

pca_explanation <- pca$sdev^2

plot_df <- plot_df |> subset(City != "Unknown")

city_pca <- ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(City)), shape = 19, size = 2) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  xlab("PC1 (6.12%)") + ylab ("PC2 (3.08%)") + labs(color = "City") + theme(legend.text = element_text(size = 14), legend.title = element_text(size = 16), axis.title.x = element_text(size = 16), axis.title.y = element_text(size = 16), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 12))

ggsave("SNP PCA City.png", device = "png", width = 15, height = 12, units = "in", dpi = 600)

city_subset_df <- plot_df |> subset(City == "Belén" | City == "Indiana" | City == "Iquitos" | City == "Punchana" | City == "San Juan Bautista")

city_subset_pca <- ggplot(city_subset_df) + theme_linedraw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(City)), shape = 19, size = 4) +
  scale_color_viridis(discrete = TRUE, option = "viridis") +
  xlab("PC1 (17.6%)") + ylab ("PC2 (8.84%)") + labs(color = "City") + theme(legend.text = element_text(size = 18), legend.title = element_text(size = 24), axis.title.x = element_text(size = 24), axis.title.y = element_text(size = 24), axis.text.x = element_text(size = 18), axis.text.y = element_text(size = 18))

ggsave("SNP PCA City Subset.png", city_subset_pca, width = 15, height = 12, units = "in", dpi = 600)

city_subset_year_pca <- ggplot(city_subset_df) + theme_linedraw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year)), shape = 19, size = 4) +
  scale_color_viridis(discrete = TRUE, option = "viridis") +
  xlab("PC1 (17.6%)") + ylab ("PC2 (8.84%)") + labs(color = "Year") + theme(legend.text = element_text(size = 18), legend.title = element_text(size = 24), axis.title.x = element_text(size = 24), axis.title.y = element_text(size = 24), axis.text.x = element_text(size = 18), axis.text.y = element_text(size = 18))

ggsave("SNP PCA City Subset by Year.png", city_subset_year_pca, width = 15, height = 12, units = "in", dpi = 600)

cityyear_subset_pca <- ggplot(city_subset_df, aes(x = PC1, y = PC2, shape = City, color = factor(Collection_Year))) + theme_linedraw() +
  geom_point(aes(group = interaction(City, Collection_Year)), size = 4) +
  scale_shape_discrete(city_subset_df$City, name = "City") +
  scale_color_viridis(discrete = TRUE, option = "viridis") +
  xlab("PC1 (17.6%)") + ylab ("PC2 (8.84%)") + labs(color = "Year") + theme(legend.text = element_text(size = 18), legend.title = element_text(size = 24), axis.title.x = element_text(size = 24), axis.title.y = element_text(size = 24), axis.text.x = element_text(size = 18), axis.text.y = element_text(size = 18))

ggsave("SNP PCA CityYear Subset.png", cityyear_subset_pca, device = "png", width = 15, height = 12, units = "in", dpi = 600)

library(patchwork)

PCA_for_supp <- city_subset_pca + city_subset_year_pca + cityyear_subset_pca +
  plot_layout(ncol = 3, nrow = 1) +
  plot_annotation(tag_levels = "A")

ggsave("PCA Supp Figure.png", PCA_for_supp, width = 30, height = 8, units = "in", dpi = 600)

ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Region))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA Region.png", device = "png", width = 15, height = 12, units = "in", dpi = 600)

ggplot(plot_df) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(State))) +
  scale_color_viridis(discrete = TRUE, option = "turbo")

ggsave("SNP PCA State.png", device = "png", width = 15, height = 12, units = "in", dpi = 600)
```

Time to get serious and make PCA plots by year within cities where we have sufficient data.

```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

BelenPCA <- plot_df |> subset(City == "Belén")

#palette("default")

ggplot(BelenPCA) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year)), size = 4) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  labs(color = "Collection Year")

ggsave("Belén PCA.png", width = 15, height = 12, units = "in", dpi = 600)

IndianaPCA <- plot_df |> subset(City == "Indiana")

palette("default")

ggplot(IndianaPCA) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year))) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  labs(color = "Collection Year")

ggsave("Indiana PCA.png", width = 15, height = 12, units = "in", dpi = 600)

IquitosPCA <- plot_df |> subset(City == "Iquitos")

palette("default")

ggplot(IquitosPCA) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year))) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  labs(color = "Collection Year")

ggsave("Iquitos PCA.png", width = 15, height = 12, units = "in", dpi = 600)

PunchanaPCA <- plot_df |> subset(City == "Punchana")

palette("default")

ggplot(PunchanaPCA) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year))) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  labs(color = "Collection Year")

ggsave("Punchana PCA.png", width = 15, height = 12, units = "in", dpi = 600)

SanJuanBautistaPCA <- plot_df |> subset(City == "San Juan Bautista")

palette("default")

ggplot(SanJuanBautistaPCA) + theme_bw() +
  geom_point(aes(x = PC1, y = PC2, color = factor(Collection_Year))) +
  scale_color_viridis(discrete = TRUE, option = "turbo") +
  labs(color = "Collection Year")

ggsave("San Juan Bautista PCA.png", width = 15, height = 12, units = "in", dpi = 600)

```


Parasitemia by Missingness at the Sample Level
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

library(dplyr)

#PvG_dp_df <- as_tibble(PvG_dp)
PvG_sample_missingness_df <- as.data.frame(PvG_sample_missingness)
PvG_sample_missingness_df <- PvG_sample_missingness_df |> rownames_to_column()
PvG_sample_missingness_df <- PvG_sample_missingness_df |> dplyr::rename("Sample" = "rowname")
PvG_sample_missingness_df <- PvG_sample_missingness_df |> dplyr::filter(!grepl("SNP", Sample))
PvG_sample_missingness_df$Sample <- PvG_sample_missingness_df$Sample |> stringr::str_remove(pattern = "-PVG")
PvG_parasitemia <- dplyr::right_join(pca_metadata, PvG_sample_missingness_df, by)
#PvG_parasitemia |> boxplot(PvG_sample_missingness ~ Asexual_Parasitemia)
#boxplot(PvG_parasitemia$PvG_sample_missingness ~ PvG_parasitemia$Asexual_Parasitemia)

##Check if WGS was done

PvG_parasitemia$WGS <- ""

PvG_parasitemia$WGS <- dplyr::case_match(PvG_parasitemia$Sample, c("MDP3364-NAMRU-1", "MDP3863-NAMRU-1", "MDP4234-NAMRU-1", "MDP4706-NAMRU-1", "MDP3487-NAMRU-1", "MDP4260-NAMRU-1", "MDP3991-NAMRU-1", "MDP6047-NAMRU-1", "MDP4278-NAMRU-1", "MDP4345-NAMRU-1", "MDP4784-NAMRU-1", "MDP3577-NAMRU-1", "MDP4824-NAMRU-1", "MDP3254-NAMRU-1", "MDP3798-NAMRU-1", "MDP3261-NAMRU-1", "MDP3286-NAMRU-1", "MDP3366-NAMRU-1", "MDP6046-NAMRU-1", "MDP3963-NAMRU-1", "MDP3488-NAMRU-1", "MDP0020-NAMRU-1", "MDP4295-NAMRU-1", "MDP6054-NAMRU-1", "MDP4184-NAMRU-1") ~ "Yes", .default = "No")

WGS_samples <- PvG_parasitemia |> subset(PvG_parasitemia$WGS == "Yes")

bad_PvG <- PvG_parasitemia |> subset(PvG_sample_missingness >= 0.05)

bad_PvG_samples <- bad_PvG$Sample

PvG_parasitemia <- PvG_parasitemia |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 10000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia != "NA") 

PvG_parasitemia |> ggplot(aes(x = bin, y = PvG_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("PvG Parasitemia Full.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

PvG_parasitemia_threshold <- PvG_parasitemia |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 1000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 10000)

PvG_parasitemia_threshold |> ggplot(aes(x = bin, y = PvG_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("PvG Parasitemia Under 10000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

PvG_parasitemia_threshold <- PvG_parasitemia_threshold |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 100, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 1000)

PvG_parasitemia_threshold |> ggplot(aes(x = bin, y = PvG_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("PvG Parasitemia Under 1000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

PvG_parasitemia_threshold <- PvG_parasitemia_threshold |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 10, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 100)

PvG_parasitemia_threshold |> ggplot(aes(x = bin, y = PvG_sample_missingness)) + geom_boxplot()

library(scales)

PvG_scatter <- PvG_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = PvG_sample_missingness)) + geom_point() + theme_linedraw() + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") + ggtitle("PvG") + scale_x_continuous(labels = label_comma())

PvG_inset <- PvG_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = PvG_sample_missingness)) + geom_point() + xlim(0, 1000)  + theme_linedraw() + theme(plot.background = element_rect(color = "black", size = 1)) + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") 

PvG_with_inset <- cowplot::ggdraw() +
  cowplot::draw_plot(PvG_scatter, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(PvG_inset, x = 0.45, y= 0.3, width = 0.5, height = 0.5)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/PvG_Parasitemia.png", PvG_with_inset, dpi = 600)

FST_sample_missingness_df <- as.data.frame(FST_sample_missingness)
FST_sample_missingness_df <- FST_sample_missingness_df |> rownames_to_column() |> rename("Sample" = "rowname")
FST_sample_missingness_df <- FST_sample_missingness_df |> dplyr::filter(!grepl("PVG", Sample))
FST_sample_missingness_df$Sample <- FST_sample_missingness_df$Sample |> stringr::str_remove(pattern = "-SNP")
FST_parasitemia <- dplyr::right_join(pca_metadata, FST_sample_missingness_df, by)
#FST_parasitemia |> boxplot(FST_sample_missingness ~ Asexual_Parasitemia)
#boxplot(FST_parasitemia$FST_sample_missingness ~ FST_parasitemia$Asexual_Parasitemia)

FST_scatter <- FST_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = FST_sample_missingness)) + geom_point() + theme_linedraw() + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") + ggtitle("PvFST") + scale_x_continuous(labels = label_comma())

FST_inset <- FST_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = FST_sample_missingness)) + geom_point() + xlim(0, 1000)  + theme_linedraw() + theme(plot.background = element_rect(color = "black", size = 1)) + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") 

FST_with_inset <- cowplot::ggdraw() +
  cowplot::draw_plot(FST_scatter, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(FST_inset, x = 0.45, y= 0.3, width = 0.5, height = 0.5)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/PvFST_Parasitemia.png", FST_with_inset, dpi = 600)

FST_parasitemia <- FST_parasitemia |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 10000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia != "NA") 

FST_parasitemia |> ggplot(aes(x = bin, y = FST_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("FST Parasitemia Full.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

FST_parasitemia_threshold <- FST_parasitemia |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 1000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 10000)

FST_parasitemia_threshold |> ggplot(aes(x = bin, y = FST_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("FST Parasitemia Under 10000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

FST_parasitemia_threshold <- FST_parasitemia_threshold |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 100, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 2000)

FST_parasitemia_threshold |> ggplot(aes(x = bin, y = FST_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("FST Parasitemia Under 2000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF5_sample_missingness_df <- as.data.frame(MAF5_sample_missingness)
MAF5_sample_missingness_df <- MAF5_sample_missingness_df |> rownames_to_column() |> dplyr::rename("Sample" = "rowname")
MAF5_sample_missingness_df <- MAF5_sample_missingness_df |> dplyr::filter(!grepl("PVG", Sample))
MAF5_sample_missingness_df$Sample <- MAF5_sample_missingness_df$Sample |> stringr::str_remove(pattern = "-SNP")
MAF5_parasitemia <- dplyr::right_join(pca_metadata, MAF5_sample_missingness_df, by)
#MAF5_parasitemia |> boxplot(MAF5_sample_missingness ~ Asexual_Parasitemia)
#boxplot(MAF5_parasitemia$MAF5_sample_missingness ~ MAF5_parasitemia$Asexual_Parasitemia)

MAF5_scatter <- MAF5_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = MAF5_sample_missingness)) + geom_point() + theme_linedraw() + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") + ggtitle("PvMAF5") + scale_x_continuous(labels = label_comma())

MAF5_inset <- MAF5_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = MAF5_sample_missingness)) + geom_point() + xlim(0, 1000)  + theme_linedraw() + theme(plot.background = element_rect(color = "black", size = 1)) + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") 

MAF5_with_inset <- cowplot::ggdraw() +
  cowplot::draw_plot(MAF5_scatter, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(MAF5_inset, x = 0.45, y= 0.3, width = 0.5, height = 0.5)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/PvMAF5_Parasitemia.png", MAF5_with_inset, dpi = 600)

MAF5_parasitemia <- MAF5_parasitemia |> dplyr::mutate(bin = cut_width(Asexual_Parasitemia, width = 10000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia != "NA") 

MAF5_parasitemia |> ggplot(aes(x = bin, y = MAF5_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF5 Parasitemia Full.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF5_parasitemia_threshold <- MAF5_parasitemia |> mutate(bin = cut_width(Asexual_Parasitemia, width = 1000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 10000)

MAF5_parasitemia_threshold |> ggplot(aes(x = bin, y = MAF5_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF5 Parasitemia Under 10000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF5_parasitemia_threshold <- MAF5_parasitemia_threshold |> mutate(bin = cut_width(Asexual_Parasitemia, width = 100, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 2000)

MAF5_parasitemia_threshold |> ggplot(aes(x = bin, y = MAF5_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF5 Parasitemia Under 2000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF40_sample_missingness_df <- as.data.frame(MAF40_sample_missingness)
MAF40_sample_missingness_df <- MAF40_sample_missingness_df |> rownames_to_column() |> rename("Sample" = "rowname")
MAF40_sample_missingness_df <- MAF40_sample_missingness_df |> dplyr::filter(!grepl("PVG", Sample))
MAF40_sample_missingness_df$Sample <- MAF40_sample_missingness_df$Sample |> stringr::str_remove(pattern = "-SNP")
MAF40_parasitemia <- dplyr::right_join(pca_metadata, MAF40_sample_missingness_df, by)
#MAF40_parasitemia |> boxplot(MAF40_sample_missingness ~ Asexual_Parasitemia)
#boxplot(MAF40_parasitemia$MAF40_sample_missingness ~ MAF40_parasitemia$Asexual_Parasitemia)


MAF40_scatter <- MAF40_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = MAF40_sample_missingness)) + geom_point() + theme_linedraw() + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") + ggtitle("PvMAF40") + scale_x_continuous(labels = label_comma())

MAF40_inset <- MAF40_parasitemia |> ggplot(aes(x = Asexual_Parasitemia, y = MAF40_sample_missingness)) + geom_point() + xlim(0, 1000)  + theme_linedraw() + theme(plot.background = element_rect(color = "black", size = 1)) + labs(y = "Sample Missingness", x = "Parasitemia") + labs(x = "Parasitemia", y = "Sample Missingness (Proportion)") 

MAF40_with_inset <- cowplot::ggdraw() +
  cowplot::draw_plot(MAF40_scatter, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(MAF40_inset, x = 0.45, y= 0.3, width = 0.5, height = 0.5)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/PvMAF40_Parasitemia.png", MAF40_with_inset, dpi = 600)

MAF40_parasitemia <- MAF40_parasitemia |> mutate(bin = cut_width(Asexual_Parasitemia, width = 10000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia != "NA") 

MAF40_parasitemia |> ggplot(aes(x = bin, y = MAF40_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF40 Parasitemia Full.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF40_parasitemia_threshold <- MAF40_parasitemia |> mutate(bin = cut_width(Asexual_Parasitemia, width = 1000, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 10000)

MAF40_parasitemia_threshold |> ggplot(aes(x = bin, y = MAF40_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF40 Parasitemia Under 10000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

MAF40_parasitemia_threshold <- MAF40_parasitemia_threshold |> mutate(bin = cut_width(Asexual_Parasitemia, width = 100, boundary = 0, dig.lab = 10)) |> subset(Asexual_Parasitemia < 2000)

MAF40_parasitemia_threshold |> ggplot(aes(x = bin, y = MAF40_sample_missingness)) + geom_boxplot() + labs(y = "Sample Missingness", x = "Parasitemia")

ggsave("MAF40 Parasitemia Under 2000.png", device = "png", dpi = 300, width = 20, height = 10, units = "in")

library(patchwork)

Parasitemia_Supp_Fig <- PvG_with_inset + FST_with_inset + MAF5_with_inset + MAF40_with_inset + plot_layout(ncol = 2, nrow = 2) + plot_annotation(tag_levels = "A")

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/Parasitemia_Supp_Fig.png", Parasitemia_Supp_Fig, dpi = 600, height = 15, width = 15, units = "in")
```

IBD
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
library(tidyverse)
library(vcfR)
library(hmmibdr)
library(sf)
library(tidygraph)
library(ggraph)
library(cowplot)

vcf <- vcfR::read.vcfR("biallelics.vcf.gz")

ad <- vcfR::extract.gt(vcf, element = "AD", as.numeric = T)
vcfR::heatmap.bp(ad)

# extract and tidy
gt <- vcfR::extract.gt(vcf, element = "GT")
gt <- dplyr::bind_cols("chrom" = vcfR::getCHROM(vcf), "pos" = vcfR::getPOS(vcf), gt)
gt <- gt %>%
  tibble::as_tibble() %>%
  tidyr::pivot_longer(cols = !c("chrom", "pos")) 

gt <- gt |>  dplyr::mutate(ishet = dplyr::case_when(value == "0/0" ~ 0,
                                         value == "0/1" ~ 1,
                                         value == "1/1" ~ 0)) 
gt <- gt |>  dplyr::group_by(name) 

gt <- gt |>  dplyr::summarise(
    propHet = mean(ishet, na.rm = TRUE))

gt |>  ggplot() +
  geom_point(aes(x = name, y = propHet)) +
  ylab("Proportion of Sample that is Het") +
  xlab("") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.75, hjust = 0.5))

mipvcf <- MIPanalyzer::vcf2mipanalyzer_biallelic(vcfR = vcf)

ibs <- MIPanalyzer::get_IBS_distance(x = mipvcf,
                                     ignore_het = FALSE,
                                     report_progress = TRUE)

colnames(ibs) <- rownames(ibs) <- colnames(vcf@gt)[2:ncol(vcf@gt)]

ibs_long <- broom::tidy(as.dist(ibs)) %>%  
  magrittr::set_colnames(c("p1", "p2", "ibsdist"))

ibs_long %>%
  ggplot() +
  geom_boxplot(aes(y = ibsdist),
               outlier.colour = "red", outlier.shape=8,
               outlier.size=4) +
  ylab("IBS Values") +
  theme_bw() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())

ibd <- MIPanalyzer::inbreeding_mle(x = mipvcf,
                                   f = seq(0.01, 0.99, 0.01),
                                   ignore_het = FALSE,
                                   report_progress = TRUE)

diag(ibd$mle) <- 1
colnames(ibd$mle) <- rownames(ibd$mle) <- colnames(vcf@gt)[2:ncol(vcf@gt)]

ibd_mle_long <- broom::tidy(as.dist(t(ibd$mle))) %>%  # note, mipanalyzer returns upper triangle
  magrittr::set_colnames(c("p1", "p2", "malecotf"))

ibd_mle_long %>%
  ggplot() +
  geom_boxplot(aes(y = malecotf),
               outlier.colour = "red", outlier.shape=8,
               outlier.size=4) +
  ylab("IBD MLE Values") +
  theme_bw() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())

mainplot <- ibd_mle_long %>%
  ggplot() +
  geom_histogram(aes(x=malecotf, y = (..count../sum(..count..))*100),
                 color = "#000000", fill = "#d9d9d9") +
  xlab("IBD") + ylab("frequency (%)") +
  theme_classic()

insetplot <- ibd_mle_long %>%
  ggplot() +
  geom_histogram(aes(x=malecotf, y = (..count../sum(..count..))*100),
                 color = "#000000", fill = "#d9d9d9") +
  xlab("IBD") + ylab("frequency (%)") +
  theme_classic() +
  coord_cartesian(xlim = c(0.5,1), ylim = c(0,3)) +
  theme_bw() +
  theme(panel.background = element_rect(fill = "transparent"),
        plot.background = element_rect(fill = "transparent"))
# cowplot
cowplot::ggdraw() +
  cowplot::draw_plot(mainplot, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(insetplot, x = 0.5, y= 0.3, width = 0.4, height = 0.4)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/IBD_cowplot.png", dpi = 600, height = 6, width = 10, units = "in")


#now time to incorporate metadata as above, but it's more complicated because there are two different samples in each

ibd_metadata_x <- pca_metadata |> dplyr::rename(p1 = Sample)

ibd_metadata_y <- pca_metadata |> dplyr::rename(p2 = Sample)

ibd_mle_long <- dplyr::left_join(ibd_mle_long, ibd_metadata_x, by = "p1") %>%
  dplyr::left_join(ibd_metadata_y, by = "p2") %>%
  rename_with(., ~gsub(".x", "_p1", .x, fixed = T)) %>%
  rename_with(., ~gsub(".y", "_p2", .x, fixed = T)) |>
  subset(Community_p1 != "Unknown") |> subset(Community_p2 != "Unknown") |> droplevels()

#Bring in COI (calculated elsewhere)
COI <-  data.table::fread("COI.txt_summary.txt", sep="\t", header=TRUE, data.table = FALSE) %>% 
  # Subset to COI results
  dplyr::filter(CorP == "C") %>%  # subset to COI information
  # select to rows we care about
  dplyr::select(-c("file", "CorP")) %>% 
  dplyr::rename(Indiv = name)

monoclonals <- COI |> subset(mean == 1)

COI2 <- COI |> subset(mean == 2)

COI3 <- COI |> subset(mean == 3)

COI4 <- COI |> subset(mean == 4)

overall_median_COI <- median(COI$mean)

COI_x <- COI |> dplyr::rename(p1 = Indiv)

COI_y <- COI |> dplyr::rename(p2 = Indiv)

ibd_mle_long <- dplyr::left_join(ibd_mle_long, COI_x, by = "p1") |>
  dplyr::left_join(COI_y, by = "p2")
ibd_mle_long <- ibd_mle_long |>  rename_with(~gsub(".x", "_p1", .x, fixed = T)) %>%
  rename_with( ~gsub(".y", "_p2", .x, fixed = T)) 

ibd_mle_long %>%
  dplyr::filter(City_p1 == City_p2) %>%
  dplyr::group_by(City_p1) %>%
  dplyr::summarise(
    withinIBD = mean(malecotf),
    COI = unique(mean_p1)) %>%
  ggplot() +
  geom_point(aes(x = withinIBD, y = COI)) +
  labs(x = "Within City IBD", y = "Simulated Mean COI",
       caption = "Remember, COI is a poor estimate of incidence. IBD is also affected by multiple factors, including randomness.") +
  theme_classic()

ibd_mle_long %>%
  ggplot() +
  geom_boxplot(aes(y = malecotf),
               outlier.colour = "red", outlier.shape=8,
               outlier.size=4) +
  ylab("IBD MLE Values") +
  theme_bw() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())

mainplot <- ibd_mle_long %>%
  ggplot() +
  geom_histogram(aes(x=malecotf, y = (..count../sum(..count..))*100),
                 color = "#000000", fill = "#d9d9d9") +
  xlab("IBD") + ylab("frequency (%)") +
  theme_classic()

insetplot <- ibd_mle_long %>%
  ggplot() +
  geom_histogram(aes(x=malecotf, y = (..count../sum(..count..))*100),
                 color = "#000000", fill = "#d9d9d9") +
  xlab("IBD") + ylab("frequency (%)") +
  theme_classic() +
  coord_cartesian(xlim = c(0.5,1), ylim = c(0,3)) +
  theme_bw() +
  theme(panel.background = element_rect(fill = "transparent"),
        plot.background = element_rect(fill = "transparent"))
# cowplot
cowplot::ggdraw() +
  cowplot::draw_plot(mainplot, x = 0, y = 0, width = 1, height = 1, scale = 1) +
  cowplot::draw_plot(insetplot, x = 0.5, y= 0.3, width = 0.4, height = 0.4)

ggsave("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/IBD_cowplot.png", dpi = 600, height = 6, width = 10, units = "in")

summary(ibd_mle_long$malecotf)

low_IBD <- ibd_mle_long |> subset(malecotf <0.25)

low_IBD_samples <- c(low_IBD$p1, low_IBD$p2) |> unique()

high_IBD <- ibd_mle_long |> subset(malecotf > 0.9)

high_IBD_samples <- c(high_IBD$p1, high_IBD$p2) |> unique()

geography <- readxl::read_xlsx("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/Peru MIP study locations.xlsx") |> dplyr::select(Community, Lat, Long)

geography_x <- geography |> dplyr::rename(Community_p1 = Community)

geography_y <- geography |> dplyr::rename(Community_p2 = Community)

ibd_mle_long <- dplyr::left_join(ibd_mle_long, geography_x, by = "Community_p1") |>
  dplyr::left_join(geography_y, by = "Community_p2") |> rename_with(~gsub(".x", "_p1", .x, fixed = T)) |>
  rename_with(~gsub(".y", "_p2", .x, fixed = T))

gc_dist <- get_spatial_distance(geography$Lat, geography$Long)

colnames(gc_dist) <- rownames(gc_dist) <- geography$Community
gcdist_long <- broom::tidy(as.dist(gc_dist)) %>%
	magrittr::set_colnames(c("Community_p1", "Community_p2", "distance")) 

ibd_mle_long_dist <- dplyr::left_join(ibd_mle_long, gcdist_long, by = c("Community_p1", "Community_p2")) %>%
  dplyr::mutate(distance = ifelse(is.na(distance), 0, distance))

ibd_mle_long_dist$distance_cat <- 
  cut(x = c(ibd_mle_long_dist$distance), 
      breaks = c(0, 1e-26, seq(40, 120, by = 40), Inf),
      right = F, 
      labels = c("Within", "40km", "80km", "120km", ">120km"))

plotdat <- ibd_mle_long_dist %>%
  dplyr::group_by(distance_cat) %>%
  dplyr::summarise(
    n = n(),
    meanIBD = mean(malecotf),
    sdIBD = sd(malecotf),
    seIBD = sdIBD/sqrt(n),
    U95CI = meanIBD + 1.96 * seIBD,
    L95CI = meanIBD - 1.96 * seIBD) 

plotdat %>% 
  ggplot() +
  geom_pointrange(aes(x = distance_cat, y = meanIBD, ymin = L95CI, ymax = U95CI)) +
  scale_size(range = c(0, 1)) +
  labs(y = "Mean IBD", x = "GC Distance", title = "Isolation by Distance") +
  theme_classic() +
  theme(plot.title = element_text(family = "Helvetica", face = "bold", hjust = 0.5, size = 14),
        axis.title = element_text(family = "Helvetica", face = "bold", hjust = 0.5, size = 12))

adj_graph <- ibd_mle_long %>%
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::mutate(IBD_community = as.factor(tidygraph::group_louvain(weights = malecotf))) |>
  ggraph::ggraph(layout = 'kk') +
  ggraph::geom_edge_link(aes(width = malecotf,
                             color = malecotf)) +
  ggraph::geom_node_point(aes(color = IBD_community),
                          size = 3) +
  ggraph::scale_edge_width_continuous(range = c(0, 1), guide = "none") +
  ggraph::geom_node_text(aes(label = name), repel = T) +
  ggraph::scale_edge_color_viridis("IBD", values = c(0,1), option = "plasma", guide = guide_colorbar(available_aes =  "edge_colour")) +
  scale_color_brewer(palette = "Set1") +
  #ggraph::theme_graph() +
  theme(legend.position = "bottom")

graph2 <- ibd_mle_long %>%
  dplyr::filter(malecotf >= 0.1) %>% 
  tidygraph::as_tbl_graph(., directed = F) %>%
  tidygraph::activate("nodes")
  dplyr::mutate(community = as.factor(tidygraph::group_louvain(weights = malecotf))) %>%
  #tidygraph::activate("edges") %>%
  ggraph::ggraph(layout = 'kk') +
  ggraph::geom_edge_link(aes(width = malecotf,
                             color = malecotf)) +
  ggraph::geom_node_point(aes(color = community),
                          size = 3) +
  ggraph::scale_edge_width_continuous(range = c(0, 1), guide = "none") +
  ggraph::geom_node_text(aes(label = name), repel = T) +
  ggraph::scale_edge_color_viridis("IBD", values = c(0,1), option = "plasma", guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_brewer(palette = "Set1") +
  ggraph::theme_graph() +
  theme(legend.position = "bottom")
  
#subset to only compare samples within cities on an individual city basis
  
#filter to only monoclonals

ibd_mle_long <- ibd_mle_long |> subset(mean_p1 == 1 & mean_p2 == 1)

network_metadata <- pca_metadata_filtered |> dplyr::select(Sample, Collection_Year)  
    
Belén_IBD <- ibd_mle_long |>
  filter(City_p1 == "Belén" & City_p2 == "Belén") |>
  dplyr::filter(malecotf >= 0.99)

mean(Belén_IBD$malecotf)

library(RColorBrewer)
getPalette <- brewer.pal(6, "Set2")
names(getPalette) <- unique(Belén_IBD$Collection_Year_p1)

Belén99 <- c(unique(Belén_IBD$p1), unique(Belén_IBD$p2)) |> unique()

Belén_network <- Belén_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.99), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Belén\n(120/269 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Belén_network

ggsave("Belén_network_0.99.png", device = png, width = 20, height = 20, units = "in", res = 600)

Indiana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Indiana" & City_p2 == "Indiana") |>
  dplyr::filter(malecotf >= 0.99)

Indiana99 <- c(unique(Indiana_IBD$p1), unique(Indiana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Indiana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Indiana_IBD$Collection_Year_p1)

Indiana_network <- Indiana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.99), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Indiana\n(3/21 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Indiana_network

ggsave("Indiana_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)

Iquitos_IBD <- ibd_mle_long |>
  filter(City_p1 == "Iquitos" & City_p2 == "Iquitos") |>
  dplyr::filter(malecotf >= 0.99)

Iquitos99 <- c(unique(Iquitos_IBD$p1), unique(Iquitos_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Iquitos_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Iquitos_IBD$Collection_Year_p1)

Iquitos_network <- Iquitos_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.99), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Iquitos\n(13/99 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Iquitos_network

ggsave("Iquitos_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)

Punchana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Punchana" & City_p2 == "Punchana") |>
  dplyr::filter(malecotf >= 0.99)

Punchana99 <- c(unique(Punchana_IBD$p1), unique(Punchana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Punchana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Punchana_IBD$Collection_Year_p1)

Punchana_network <- Punchana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.99), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Punchana\n(23/94 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Punchana_network

ggsave("Punchana_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)

San_Juan_Bautista_IBD <- ibd_mle_long |>
  filter(City_p1 == "San Juan Bautista" & City_p2 == "San Juan Bautista") |>
  dplyr::filter(malecotf >= 0.99)

SJB99 <- c(unique(San_Juan_Bautista_IBD$p1), unique(San_Juan_Bautista_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(San_Juan_Bautista_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(San_Juan_Bautista_IBD$Collection_Year_p1)

San_Juan_Bautista_network <- San_Juan_Bautista_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.99), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("San Juan Bautista\n(51/186 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

San_Juan_Bautista_network

ggsave("San_Juan_Bautista_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)


#plot_grid(Belén_network, Indiana_network, Iquitos_network, Punchana_network, San_Juan_Bautista_network, align = "hv")

library(patchwork)

design <- "
  AAAABBCC
  AAAABBCC
  AAAABBCC
  AAAADDEE
  AAAADDEE
  AAAADDEE
  FFFFFFFF
"

Fig2_IBD <- Belén_network + Indiana_network + Iquitos_network + Punchana_network + San_Juan_Bautista_network + guide_area() +
  plot_layout(design = design, guides = "collect") & theme (legend.position = "bottom") & 
  scale_color_manual(values = getPalette, name = "Year", limits = c(2012, 2013, 2014, 2015, 2016, 2017))

ggsave("IBD_networks.png", Fig2_IBD, device = "png", dpi = 600, width = 30, height = 20, units = "in") 

```

Supplemental IBD Figures - 0.25, 0.5, 0.9 cutoffs

```{r}
Belén_IBD <- ibd_mle_long |>
  filter(City_p1 == "Belén" & City_p2 == "Belén") |>
  dplyr::filter(malecotf >= 0.25)

mean(Belén_IBD$malecotf)

library(RColorBrewer)
getPalette <- brewer.pal(6, "Set2")
names(getPalette) <- unique(Belén_IBD$Collection_Year_p1)

Belén25 <- c(unique(Belén_IBD$p1), unique(Belén_IBD$p2)) |> unique()

Belén_network <- Belén_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.25), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Belén\n(226/269 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Belén_network

ggsave("Belén_network_0.25.png", device = png, width = 20, height = 20, units = "in", res = 600)

Indiana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Indiana" & City_p2 == "Indiana") |>
  dplyr::filter(malecotf >= 0.25)

Indiana25 <- c(unique(Indiana_IBD$p1), unique(Indiana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Indiana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Indiana_IBD$Collection_Year_p1)

Indiana_network <- Indiana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.25), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Indiana\n(13/21 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Indiana_network

ggsave("Indiana_network_0.25.png", device = png, width = 10, height = 10, units = "in", res = 600)

Iquitos_IBD <- ibd_mle_long |>
  filter(City_p1 == "Iquitos" & City_p2 == "Iquitos") |>
  dplyr::filter(malecotf >= 0.25)

Iquitos25 <- c(unique(Iquitos_IBD$p1), unique(Iquitos_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Iquitos_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Iquitos_IBD$Collection_Year_p1)

Iquitos_network <- Iquitos_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.25), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Iquitos\n(78/99 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Iquitos_network

ggsave("Iquitos_network_0.25.png", device = png, width = 10, height = 10, units = "in", res = 600)

Punchana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Punchana" & City_p2 == "Punchana") |>
  dplyr::filter(malecotf >= 0.25)

Punchana25 <- c(unique(Punchana_IBD$p1), unique(Punchana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Punchana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Punchana_IBD$Collection_Year_p1)

Punchana_network <- Punchana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.25), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Punchana\n(79/94 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Punchana_network

ggsave("Punchana_network_0.25.png", device = png, width = 10, height = 10, units = "in", res = 600)

San_Juan_Bautista_IBD <- ibd_mle_long |>
  filter(City_p1 == "San Juan Bautista" & City_p2 == "San Juan Bautista") |>
  dplyr::filter(malecotf >= 0.25)

SJB25 <- c(unique(San_Juan_Bautista_IBD$p1), unique(San_Juan_Bautista_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(San_Juan_Bautista_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(San_Juan_Bautista_IBD$Collection_Year_p1)

San_Juan_Bautista_network <- San_Juan_Bautista_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.25), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.25, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("San Juan Bautista\n(162/186 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

San_Juan_Bautista_network

ggsave("San_Juan_Bautista_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)


#plot_grid(Belén_network, Indiana_network, Iquitos_network, Punchana_network, San_Juan_Bautista_network, align = "hv")

library(patchwork)

design <- "
  AAAABBCC
  AAAABBCC
  AAAABBCC
  AAAADDEE
  AAAADDEE
  AAAADDEE
  FFFFFFFF
"

IBD25 <- Belén_network + Indiana_network + Iquitos_network + Punchana_network + San_Juan_Bautista_network + guide_area() +
  plot_layout(design = design, guides = "collect") & theme (legend.position = "bottom") & 
  scale_color_manual(values = getPalette, name = "Year", limits = c(2012, 2013, 2014, 2015, 2016, 2017))

ggsave("IBD_0.25_networks.png", IBD25, device = "png", dpi = 600, width = 30, height = 20, units = "in")

Belén_IBD <- ibd_mle_long |>
  filter(City_p1 == "Belén" & City_p2 == "Belén") |>
  dplyr::filter(malecotf >= 0.50)

mean(Belén_IBD$malecotf)

library(RColorBrewer)
getPalette <- brewer.pal(6, "Set2")
names(getPalette) <- unique(Belén_IBD$Collection_Year_p1)

Belén50 <- c(unique(Belén_IBD$p1), unique(Belén_IBD$p2)) |> unique()

Belén_network <- Belén_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.50), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.50, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Belén\n(211/269 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Belén_network

ggsave("Belén_network_0.50.png", device = png, width = 20, height = 20, units = "in", res = 600)

Indiana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Indiana" & City_p2 == "Indiana") |>
  dplyr::filter(malecotf >= 0.50)

Indiana50 <- c(unique(Indiana_IBD$p1), unique(Indiana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Indiana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Indiana_IBD$Collection_Year_p1)

Indiana_network <- Indiana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.50), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.50, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Indiana\n(8/21 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Indiana_network

ggsave("Indiana_network_0.50.png", device = png, width = 10, height = 10, units = "in", res = 600)

Iquitos_IBD <- ibd_mle_long |>
  filter(City_p1 == "Iquitos" & City_p2 == "Iquitos") |>
  dplyr::filter(malecotf >= 0.50)

Iquitos50 <- c(unique(Iquitos_IBD$p1), unique(Iquitos_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Iquitos_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Iquitos_IBD$Collection_Year_p1)

Iquitos_network <- Iquitos_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.50), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.50, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Iquitos\n(46/99 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Iquitos_network

ggsave("Iquitos_network_0.50.png", device = png, width = 10, height = 10, units = "in", res = 600)

Punchana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Punchana" & City_p2 == "Punchana") |>
  dplyr::filter(malecotf >= 0.50)

Punchana50 <- c(unique(Punchana_IBD$p1), unique(Punchana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Punchana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Punchana_IBD$Collection_Year_p1)

Punchana_network <- Punchana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.50), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.50, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Punchana\n(66/94 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Punchana_network

ggsave("Punchana_network_0.50.png", device = png, width = 10, height = 10, units = "in", res = 600)

San_Juan_Bautista_IBD <- ibd_mle_long |>
  filter(City_p1 == "San Juan Bautista" & City_p2 == "San Juan Bautista") |>
  dplyr::filter(malecotf >= 0.50)

SJB50 <- c(unique(San_Juan_Bautista_IBD$p1), unique(San_Juan_Bautista_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(San_Juan_Bautista_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(San_Juan_Bautista_IBD$Collection_Year_p1)

San_Juan_Bautista_network <- San_Juan_Bautista_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.50), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.50, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("San Juan Bautista\n(137/186 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

San_Juan_Bautista_network

ggsave("San_Juan_Bautista_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)


#plot_grid(Belén_network, Indiana_network, Iquitos_network, Punchana_network, San_Juan_Bautista_network, align = "hv")

library(patchwork)

design <- "
  AAAABBCC
  AAAABBCC
  AAAABBCC
  AAAADDEE
  AAAADDEE
  AAAADDEE
  FFFFFFFF
"

IBD50 <- Belén_network + Indiana_network + Iquitos_network + Punchana_network + San_Juan_Bautista_network + guide_area() +
  plot_layout(design = design, guides = "collect") & theme (legend.position = "bottom") & 
  scale_color_manual(values = getPalette, name = "Year", limits = c(2012, 2013, 2014, 2015, 2016, 2017))

ggsave("IBD_0.50_networks.png", IBD50, device = "png", dpi = 600, width = 30, height = 20, units = "in")

Belén_IBD <- ibd_mle_long |>
  filter(City_p1 == "Belén" & City_p2 == "Belén") |>
  dplyr::filter(malecotf >= 0.90)

mean(Belén_IBD$malecotf)

library(RColorBrewer)
getPalette <- brewer.pal(6, "Set2")
names(getPalette) <- unique(Belén_IBD$Collection_Year_p1)

Belén90 <- c(unique(Belén_IBD$p1), unique(Belén_IBD$p2)) |> unique()

Belén_network <- Belén_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.90), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.90, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Belén\n(192/269 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Belén_network

ggsave("Belén_network_0.90.png", device = png, width = 20, height = 20, units = "in", res = 600)

Indiana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Indiana" & City_p2 == "Indiana") |>
  dplyr::filter(malecotf >= 0.90)

Indiana90 <- c(unique(Indiana_IBD$p1), unique(Indiana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Indiana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Indiana_IBD$Collection_Year_p1)

Indiana_network <- Indiana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.90), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.90, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Indiana\n(6/21 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Indiana_network

ggsave("Indiana_network_0.90.png", device = png, width = 10, height = 10, units = "in", res = 600)

Iquitos_IBD <- ibd_mle_long |>
  filter(City_p1 == "Iquitos" & City_p2 == "Iquitos") |>
  dplyr::filter(malecotf >= 0.90)

Iquitos90 <- c(unique(Iquitos_IBD$p1), unique(Iquitos_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Iquitos_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Iquitos_IBD$Collection_Year_p1)

Iquitos_network <- Iquitos_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.90), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.90, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Iquitos\n(27/99 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Iquitos_network

ggsave("Iquitos_network_0.90.png", device = png, width = 10, height = 10, units = "in", res = 600)

Punchana_IBD <- ibd_mle_long |>
  filter(City_p1 == "Punchana" & City_p2 == "Punchana") |>
  dplyr::filter(malecotf >= 0.90)

Punchana90 <- c(unique(Punchana_IBD$p1), unique(Punchana_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(Punchana_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(Punchana_IBD$Collection_Year_p1)

Punchana_network <- Punchana_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.90), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.90, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("Punchana\n(46/94 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom") +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

Punchana_network

ggsave("Punchana_network_0.90.png", device = png, width = 10, height = 10, units = "in", res = 600)

San_Juan_Bautista_IBD <- ibd_mle_long |>
  filter(City_p1 == "San Juan Bautista" & City_p2 == "San Juan Bautista") |>
  dplyr::filter(malecotf >= 0.90)

SJB90 <- c(unique(San_Juan_Bautista_IBD$p1), unique(San_Juan_Bautista_IBD$p2)) |> unique()

#getPalette <- brewer.pal(dplyr::n_distinct(San_Juan_Bautista_IBD$Collection_Year_p1), "Set2")
#names(getPalette) <- unique(San_Juan_Bautista_IBD$Collection_Year_p1)

San_Juan_Bautista_network <- San_Juan_Bautista_IBD |>
  tidygraph::as_tbl_graph(., directed = F) |>
  tidygraph::activate("nodes") |>
  dplyr::left_join(network_metadata, by = c("name" = "Sample")) |>
  ggraph::ggraph("stress", bbox = 15) +
  ggraph::geom_edge_link(aes(width = 0.90), show.legend = FALSE) +
  #ggraph::geom_node_point() +
  ggraph::geom_node_point(aes(color = factor(Collection_Year)), size = 10) +
  ggraph::scale_edge_width_continuous(range = c(0.90, 1), guide = "none") +
  #ggraph::geom_node_text(aes(label = name), repel = T, size = 6) +
  #ggraph::scale_edge_color_viridis("IBD", begin = 0.5, end = 1, option = "plasma",
  #                                 guide = guide_colorbar(available_aes = "edge_colour")) +
  scale_color_manual(values = getPalette,
                     name = "Year") +
  guides(color = guide_legend(nrow = 1)) +
  ggtitle("San Juan Bautista\n(111/186 samples)") +
  ggraph::theme_graph(base_size = 20) +
  theme(legend.position = "bottom", plot.title = element_text(size = 48), legend.text = element_text(size = 36), legend.title = element_text(size = 40), plot.background = element_rect(color = "black", linewidth = 2)) 

San_Juan_Bautista_network

ggsave("San_Juan_Bautista_network_0.99.png", device = png, width = 10, height = 10, units = "in", res = 600)


#plot_grid(Belén_network, Indiana_network, Iquitos_network, Punchana_network, San_Juan_Bautista_network, align = "hv")

library(patchwork)

design <- "
  AAAABBCC
  AAAABBCC
  AAAABBCC
  AAAADDEE
  AAAADDEE
  AAAADDEE
  FFFFFFFF
"

IBD90 <- Belén_network + Indiana_network + Iquitos_network + Punchana_network + San_Juan_Bautista_network + guide_area() +
  plot_layout(design = design, guides = "collect") & theme (legend.position = "bottom") & 
  scale_color_manual(values = getPalette, name = "Year", limits = c(2012, 2013, 2014, 2015, 2016, 2017))

ggsave("IBD_0.90_networks.png", IBD90, device = "png", dpi = 600, width = 30, height = 20, units = "in")
```


Allele frequencies (from Sean's code)

#You need to manually edit the CSVs in the PvG analysis folder to standardize things prior to running any of this code. Specifically, I did this to get rid of the non-standard sample names and remove the samples with no data.
#Also of note for this one is that the file names are weirdly inconsistent. They're missing the "_AA" designation, so I added that to make things easier downstream.
#Annoyingly, the gene names aren't included even though I thought I fixed that so I had to add some code to add them.
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
# load packages 
library(miplicorn)
library(knitr)
library(kableExtra)
library(readr)
library(tidyverse)
library(tidyr)
library(gt)
library(webshot2)
library(writexl)
library(cowplot)
# set seed
set.seed(1)

extract_sampleID <- function(x) {
  # split all three by delimiter
  splt <- str_split_fixed(x,'-',4)
  # make a dataframe 
  splt_ID <- as.data.frame(splt) %>% 
    mutate(ID = if_else(
      V1 == 'NAMRU',.[[2]],
      .[[1]]))
  return(splt_ID$ID)
}

# Make a list of amino acids
amino_acids <- vector(mode="list", length=18)
names(amino_acids) <- c("Val", "Ala", "His","Leu","Arg","Pro","Lys","Thr","Cys","Met","Tyr","Ile","Asn","Glu","Ser","Gly","Asp","Phe")
amino_acids[[1]] <- "V"; amino_acids[[2]] <- "A"; amino_acids[[3]] <- "H"; amino_acids[[4]] <- "L"; amino_acids[[5]] <- "R"; amino_acids[[6]] <- "P"; amino_acids[[7]] <- "K"; amino_acids[[8]] <- "T"; amino_acids[[9]] <- "C"; amino_acids[[10]] <- "M"; amino_acids[[11]] <- "Y"; amino_acids[[12]] <- "I"; amino_acids[[13]] <- "N"; amino_acids[[14]] <- "E"; amino_acids[[15]] <- "S"; amino_acids[[16]] <- "G"; amino_acids[[17]] <- "D"; amino_acids[[18]] <- "F"

nonpara <- function(x,iters) {
  # make as dataframe
  final <- matrix(data = NA, nrow = 1, ncol = 3)
  final <- as.data.frame(final)
  colnames(final) <- c('mean','L95','U95')
  nonparam_mean <- rep(NA, iters)
  for (q in 1:iters) {
    nonparam_mean[q] <- mean(sample(x, length(x), replace = T)) 
    }
  mean <- mean(nonparam_mean)
  final[1,1] <-round(mean,3)
  nonparamL95 <- quantile(nonparam_mean, probs = 0.025, na.rm = TRUE)
  final[1,2] <- round(nonparamL95,3)
  nonparamU95 <- quantile(nonparam_mean, probs = 0.975, na.rm = TRUE)
  final[1,3] <- round(nonparamU95,3)
  return(final)
}

calculate_freq <- function(x) {
  x_freq <- x %>%
    dplyr::mutate(freq = (alt_umi_count/(alt_umi_count+ref_umi_count)))
  freq_table <- x_freq %>%
    group_by(mutation_name) %>%
    do(nonpara(.$freq,iters = 1e3))
  sample_size <- x_freq %>% 
    group_by(mutation_name) %>% 
    summarise(n = n())
  freq_table <- left_join(freq_table,sample_size,by = "mutation_name")
  return(freq_table)
}

ref_file <- "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/reference_AA_table.csv"
alt_file <- "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/alternate_AA_table.csv"
cov_file <- "C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/coverage_AA_table.csv"

aa_data <- miplicorn::read_tbl_ref_alt_cov(ref_file,
                                        alt_file,
                                        cov_file)

aa_data <- aa_data |> dplyr::mutate(gene = dplyr::case_when(chrom == "PvP01_09_v1" & pos >= 1458201 & pos <= 1460984 ~ "ama1",
                                                                  chrom == "PvP01_01_v1" & pos >= 440364 & pos <= 446544 ~ "crt",
                                                               chrom == "PvP01_08_v1" & pos >= 1508714 & pos <= 1509805 ~ "csp",
                                                               chrom == "PvP01_06_v1" & pos >= 981059 & pos <= 987037 ~ "dbp",
                                                               chrom == "PvP01_01_v1" & pos >= 104013 & pos <= 107429 ~ "ebp",
                                                               chrom == "PvP01_05_v1" & pos >= 1075584 & pos <= 1079966 ~ "dhfr-ts",
                                                               chrom == "PvP01_05_v1" & pos >= 252986 & pos <= 256400 ~ "gama",
                                                               chrom == "PvP01_10_v1" & pos >= 476677 & pos <= 483934 ~ "mdr1",
                                                               chrom == "PvP01_07_v1" & pos >= 1215432 & pos <= 1220636 ~ "msp1",
                                                               chrom == "PvP01_14_v1" & pos >= 1269756 & pos <= 1272657 ~ "pppk-dhps",
                                                               chrom == "PvP01_07_v1" & pos >= 71106 & pos <= 80980 ~ "rbp1a",
                                                               chrom == "PvP01_07_v1" & pos >= 59612 & pos <= 70478 ~ "rbp1b",
                                                               chrom == "PvP01_14_v1" & pos >= 112686 & pos <= 121381 ~ "rbp2a",
                                                            chrom == "PvP01_08_v1" & pos >= 33312 & pos <= 43704 ~ "rbp2b",
                                                            chrom == "PvP01_05_v1" & pos >= 1458611 & pos <= 1467388 ~ "rbp2c",
                                                            chrom == "PvP01_05_v1" & pos >= 1390201 & pos <= 1392765 ~ "trag11",
                                                            chrom == "PvP01_11_v1" & pos >= 68505 & pos <= 71449 ~ "trag19",
                                                            chrom == "PvP01_14_v1"& pos >= 81068 & pos <= 82841 ~ "trag21",
                                                            chrom == "PvP01_14_v1" & pos >= 2984025 & pos <= 2988536 ~ "trag22",
                                                            chrom == "Transfer.PvP01_00_1.final" & pos >= 39887 & pos <= 40931 ~ "trag26",
                                                            chrom == "Transfer.PvP01_00_1.final" & pos >= 72661 & pos <= 75073 ~ "trag32",
                                                            chrom == "PvP01_07_v1" & pos >= 45445 & pos >= 46410 ~ "trag34",
                                                            chrom == "PvP01_09_v1" & pos >= 2174904 & pos >= 2176967 ~  "trag36_1",
                                                            chrom == "Transfer.PvP01_00_1.final" & pos >= 57969 & pos <= 58929 ~ "trag36_2",
                                                            chrom == "PvP01_05_v1" & pos >= 150938 & pos <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))                                        
                                        
aa_data <- aa_data |> subset(gene != "not_targeted")

# merge some metadata columns
aa_data <- dplyr::left_join(aa_data, pca_metadata_filtered, by = c("sample" = "Sample")) |> subset(Community != "Unknown") |> droplevels()

aa_data$sample <- extract_sampleID(aa_data$sample)

# filter coverage less than 5
aa_data <- aa_data %>% 
  dplyr::filter(!(coverage < 5))

# Add Metadata
#metadata <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_full_run/Combined_Analysis/metadata.csv", data.table = FALSE, header = TRUE)

aa_data <- aa_data |> mutate (mutation_name = paste0(gene, "-", aa_change))

for(i in names(amino_acids)) {
  aa_data$mutation_name <- str_replace_all(aa_data$mutation_name,i,amino_acids[[i]])
} 

data_split <- aa_data |> split(aa_data$gene)

list2env(data_split,envir=.GlobalEnv)

freq_split <- lapply(data_split, function(i) calculate_freq(i) |> filter(!(mean < 0.005))) 

names(freq_split) <- paste0(names(freq_split),"_freq")

list2env(freq_split, envir=.GlobalEnv)

ggplot(ama1_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("ama1.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(crt_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("crt.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(csp_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("csp.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(dbp_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("dbp.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(ebp_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("ebp.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(`dhfr-ts_freq`, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("dhfr-ts.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(gama_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("gama.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(mdr1_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("mdr1.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(msp1_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("msp1.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(`pppk-dhps_freq`, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("pppk-dhps.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(rbp1a_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("rbp1a.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(rbp1b_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("rbp1b.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(rbp2a_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("rbp2a.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(rbp2b_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("rbp2b.png",device = png, width = 24, height = 20, units = 'in', res = 600)

ggplot(rbp2c_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("rbp2c.png",device = png, width = 36, height = 20, units = 'in', res = 600)

trag_freq <- rbind(trag11_freq, trag19_freq, trag21_freq, trag22_freq, trag26_freq, trag32_freq, trag34_freq, trag36_2_freq, trag38_freq)

ggplot(trag_freq, aes(x = mutation_name, y = mean)) + 
  geom_bar(stat = 'identity',position=position_dodge(), width =.4)+ 
  coord_cartesian(ylim=c(0,1)) +
  theme_classic() +
  labs(y= "Frequency", 
       #title = "Prevalence Based On Region"
       ) +
  #guides(fill=guide_legend(title="Region")) +
  geom_errorbar(aes(ymin=L95, ymax=U95), width=.5,
                 position=position_dodge(.4)) +
  scale_fill_manual(values = c("#2538be")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 25),
        axis.title = element_text(size = 35),
        legend.text = element_text(size = 25),
        legend.title = element_text(size = 25))

ggsave("trag.png",device = png, width = 24, height = 20, units = 'in', res = 600)

library(kableExtra)

library(flextable)
border_style = officer::fp_border(color="black", width=3)
# arrange mutations by factor
#mutations_for_tbl <- c("crt-K76T","crt-M74I","crt-N75E", "dhfr-ts-S108N","dhfr-ts-N51I","dhfr-ts-C59R","dhps-A437G","dhfr-ts-I164L","dhfr-ts-S108T","dhps-A581G", "dhps-K540E", "k13-K189N", "k13-K189T", "mdr1-N86Y", "mdr1-D1246Y", "mdr1-Y184F","mdr2-I492V","dhfr-ts-A16V")

#freq_main_zan_tbl_present <- freq_main_zan_tbl %>% 
#  filter(mutation_name %in% mutations_for_tbl)

drug_resistance <- rbind(crt_freq, `dhfr-ts_freq`, mdr1_freq, `pppk-dhps_freq`)

vaccine_targets <- rbind(ama1_freq, csp_freq, dbp_freq, ebp_freq, gama_freq, msp1_freq, trag_freq, rbp1a_freq, rbp1b_freq, rbp2a_freq, rbp2b_freq, rbp2c_freq)

drug_resistance_table <- drug_resistance |>  mutate(across(everything(), round, 3)) |> 
  unite(CI,L95,U95,remove = T, sep = "-") |>
  
  flextable::flextable() %>% 
  flextable::autofit() %>% 
  #flextable::width(j=1, width = 2) %>% 
  #flextable::width(j=c(2,4), width = 1) %>% 
  #flextable::width(j=c(3,5), width = 1.5) %>% 
  #flextable::add_header_row(
  #  top = TRUE,                
  #  values = c("Mutation",
  #             "",
   #            "")) |> 
  flextable::set_header_labels(
    mutation_name = "Mutation",
    mean = "Mutant Allele Prevalence‡",
    CI = "CI‡",
    n = "# Genotyped Samples§") |>
  flextable::autofit() |>
  flextable::align(align = "center", part = "all") |>
  # merge mutation name header
 # flextable::merge_at(i = 1:2, j = 1, part = "header") %>% 
  # merge Zanzibar part
  #flextable::merge_at(i = 1, j = 2:4, part = "header") %>%
  # merge Mainland part
  #flextable::merge_at(i = 1, j = 5:7, part = "header") %>%
  # Remove all existing borders
  flextable::border_remove() %>% 
  # add  box theme
  flextable::theme_box() %>%
  # change borders
  border(border = border_style) %>% 
  border(border = border_style, part = "header") |>
  flextable::align(align = "center", part = "all")#%>% 
  #flextable::align(align = "center", j = c(2:7), part = "all")

#library(webshot2)
#save_as_image(drug_resistance_table, path = "drug_resistance.pdf",webshot = "webshot2")
save_as_docx(drug_resistance_table, path = "drug_resistance.docx")

vaccine_targets_table <- vaccine_targets |>  mutate(across(everything(), round, 3)) |> 
  unite(CI,L95,U95,remove = T, sep = "-") |>
  
  flextable::flextable() %>% 
  flextable::autofit() %>% 
  #flextable::width(j=1, width = 2) %>% 
  #flextable::width(j=c(2,4), width = 1) %>% 
  #flextable::width(j=c(3,5), width = 1.5) %>% 
  #flextable::add_header_row(
  #  top = TRUE,                
  #  values = c("Mutation",
  #             "",
   #            "")) |> 
  flextable::set_header_labels(
    mutation_name = "Mutation",
    mean = "Mutant Allele Prevalence‡",
    CI = "CI‡",
    n = "# Genotyped Samples§") |>
  flextable::autofit() |>
  flextable::align(align = "center", part = "all") |>
  # merge mutation name header
 # flextable::merge_at(i = 1:2, j = 1, part = "header") %>% 
  # merge Zanzibar part
  #flextable::merge_at(i = 1, j = 2:4, part = "header") %>%
  # merge Mainland part
  #flextable::merge_at(i = 1, j = 5:7, part = "header") %>%
  # Remove all existing borders
  flextable::border_remove() %>% 
  # add  box theme
  flextable::theme_box() %>%
  # change borders
  border(border = border_style) %>% 
  border(border = border_style, part = "header") |>
  flextable::align(align = "center", part = "all")#%>% 
  #flextable::align(align = "center", j = c(2:7), part = "all")

#library(webshot2)
#save_as_image(vaccine_targets_table, path = "vaccine_targets.pdf",webshot = "webshot2")
save_as_docx(vaccine_targets_table, path = "vaccine_targets.docx")

#MalariaGEN frequencies

#this was such a pain but eventually I figured it out using the code from above to get the positions to check the frequencies of mutations that we detected

mutations <- c("crt-K9dup", "dhfr-ts-F57L", "dhfr-ts-S58R", "dhfr-ts-N117S", "dhfr-ts-I173L", "pppk-dhps-G383A", "mdr1-S698G", "mdr1-L908M", "mdr1-M958T", "mdr1-F976Y", "mdr1-L1076F")

mutation_data <- aa_data |> dplyr::filter(mutation_name %in% mutations) |> dplyr::select("chrom", "pos", "ref", "alt", "mutation_name") |> dplyr::distinct()

mutation_data <- mutation_data |> dplyr::rename(CHROM = chrom, POS = pos)

MG_freq <- dplyr::left_join(mutation_data, MG_AF)

#find nucleotide positions of the ones we didn't detect

#do a query on a per-gene basis to the malariagen VCF

#bcftools query -f '%CHROM\t%POS\t%REF\t%ALT{0}\t%AF\t%SNPEFF_AMINO_ACID_CHANGE[\t%AD{0}]\n' --regions PvP01_05_v1:1075584-1079966 --include 'FILTER="PASS" && TYPE="snp" && N_ALT=1' --print-header malariagen_DR.vcf.gz > dhfr_AF.txt

dhfr_af <- data.table::fread("dhfr_AF.txt") #then get AF from AF column

dhps_af <- data.table::fread("dhps_AF.txt")

mdr1_af <- data.table::fread("mdr1_AF.txt")

#Tajima's D - calculated in vcf-kit in bins of 10 and visualized here

#had to get rid of gaps as above to get this to work a second time    

#vk tajima 10 10 ../PvG_wrangler/analysis/variants.vcf.gz > PvG_vk_Tajima.txt

PvG_Tajima <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/PvG_vk_Tajima.txt", data.table = FALSE, header = TRUE) #|> subset(CHROM != "Transfer.PvP01_00_1.final")

PvG_Tajima <- PvG_Tajima |> dplyr::mutate(gene = dplyr::case_when(CHROM == "PvP01_09_v1" & BIN_START >= 1458201 & BIN_END <= 1460984 ~ "ama1",
                                                                  CHROM == "PvP01_01_v1" & BIN_START >= 440364 & BIN_END <= 446544 ~ "crt",
                                                               CHROM == "PvP01_08_v1" & BIN_START >= 1508714 & BIN_END <= 1509805 ~ "csp",
                                                               CHROM == "PvP01_06_v1" & BIN_START >= 981059 & BIN_END <= 987037 ~ "dbp",
                                                               CHROM == "PvP01_01_v1" & BIN_START >= 104013 & BIN_END <= 107429 ~ "ebp",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 1075584 & BIN_END <= 1079966 ~ "dhfr-ts",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 252986 & BIN_END <= 256400 ~ "gama",
                                                               CHROM == "PvP01_10_v1" & BIN_START >= 476677 & BIN_END <= 483934 ~ "mdr1",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 1215432 & BIN_END <= 1220636 ~ "msp1",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 1269756 & BIN_END <= 1272657 ~ "pppk-dhps",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 71106 & BIN_END <= 80980 ~ "rbp1a",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 59612 & BIN_END <= 70478 ~ "rbp1b",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 112686 & BIN_END <= 121381 ~ "rbp2a",
                                                            CHROM == "PvP01_08_v1" & BIN_START >= 33312 & BIN_END <= 43704 ~ "rbp2b",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1458611 & BIN_END <= 1467388 ~ "rbp2c",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1390201 & BIN_END <= 1392765 ~ "trag11",
                                                            CHROM == "PvP01_11_v1" & BIN_START >= 68505 & BIN_END <= 71449 ~ "trag19",
                                                            CHROM == "PvP01_14_v1"& BIN_START >= 81068 & BIN_END <= 82841 ~ "trag21",
                                                            CHROM == "PvP01_14_v1" & BIN_START >= 2984025 & BIN_END <= 2988536 ~ "trag22",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 39887 & BIN_END <= 40931 ~ "trag26",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 72661 & BIN_END <= 75073 ~ "trag32",
                                                            CHROM == "PvP01_07_v1" & BIN_START >= 45445 & BIN_START >= 46410 ~ "trag34",
                                                            CHROM == "PvP01_09_v1" & BIN_START >= 2174904 & BIN_START >= 2176967 ~  "trag36_1",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 57969 & BIN_END <= 58929 ~ "trag36_2",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 150938 & BIN_END <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))

PvG_Tajima <- PvG_Tajima |> subset(gene != "not_targeted") |> dplyr::mutate(POS = BIN_END - (BIN_END - BIN_START)/2)

summary(PvG_Tajima$TajimaD)

#transfers <- PvG_Tajima |> subset(CHROM == "Transfer.PvP01_00_1.final")

#PvG_Tajima <- PvG_Tajima |> subset(CHROM != "Transfer.PvP01_00_1.final")

#PvG_Tajima$CHROM <- PvG_Tajima$CHROM |> stringr::str_remove(pattern = "PvP01_") |> stringr::str_remove(pattern = "_v1") |> as.numeric((PvG_Tajima$CHROM))

#PvG_Tajima <- PvG_Tajima |> dplyr::mutate(GEN_POS = paste0(CHROM,BIN_START))

#PvG_Tajima <- PvG_Tajima |> dplyr::mutate(GENE = dplyr::case_when(GEN_POS >= 1104013 & GEN_POS <= 1107429 ~ "DBP2",
#                                                                  GEN_POS >= 1440364 & GEN_POS <= 1446544 ~ "CRT",
#                                                                  GEN_POS >= 51075584 & GEN_POS <= 51079966 ~ "DHFR-TS",
#                                                                  GEN_POS >= 51458611 & GEN_POS <= 51467388 ~ "RBP2c",
#                                                                  GEN_POS >= 5252986 & GEN_POS <= 5256400 ~ "GAMA",
#                                                                  GEN_POS >= 51390201 & GEN_POS <= 51392765 ~ "TRAG11",
#                                                                  GEN_POS >= 5150938 & GEN_POS <= 5152029 ~ "TRAG38",
#                                                                  GEN_POS >= 6981059 & GEN_POS <= 6987037 ~ "DBP",
#                                                                  GEN_POS >= 71215432 & GEN_POS <= 71220636 ~ "MSP1",
#                                                                  GEN_POS >= 771106 & GEN_POS <= 780980 ~ "RBP1a",
#                                                                  GEN_POS >= 759612 & GEN_POS <= 770478 ~ "RBP1b",
#                                                                  GEN_POS >= 745445 & GEN_POS <= 746410 ~ "TRAG34",
#                                                                  GEN_POS >= 833312 & GEN_POS <= 843704 ~ "RBP2b",
#                                                                  GEN_POS >= 81508714 & GEN_POS <= 81509805 ~ "CSP",
#                                                                  GEN_POS >= 91458201 & GEN_POS <= 91460984 ~ "AMA1",
#                                                                  GEN_POS >= 92174904 & GEN_POS <= 92176967 ~ "TRAG36",
#                                                                  GEN_POS >= 10476677 & GEN_POS <= 10483934 ~ "MDR1",
#                                                                  GEN_POS >= 1168505 & GEN_POS <= 1171449 ~ "TRAG19",
#                                                                  GEN_POS >= 14112686 & GEN_POS <= 14121381 ~ "RBP2a",
#                                                                  GEN_POS >= 141269756 & GEN_POS <= 141272657 ~ "PPPK-DHPS",
#                                                                  GEN_POS >= 1481068 & GEN_POS <= 1482841 ~ "TRAG21",
#                                                                  GEN_POS >= 142984025 & GEN_POS <= 142988536 ~ "TRAG22"))

#remotes::install_github("hemstrow/snpR", build_vignettes = T, build_opts = c("--no-resave-data", "--no-manual")) # windows

#library(snpR)

#options(ggrepel.max.overlaps = Inf) #this does not result in a beautiful plot, but it does ensure that every significant point is labeled

#plot_manhattan(PvG_Tajima, chr = "CHROM", bp = "BIN_START", plot_var = "TajimaD", significant = abs(2), snp = "GENE")

#ggsave(filename = "Tajima Manhattan All Genes but trag26 trag32 trag36_2.png", dpi = 600, width = 20, height = 10, units = "in")

Tajima_split <- PvG_Tajima |> split(PvG_Tajima$gene)

names(Tajima_split) <- paste0(names(Tajima_split),"_Tajima")

list2env(Tajima_split,envir=.GlobalEnv)

missing_coords <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/PvG_missing_ranges_edited.csv", header = T, data.table = F) #gave up on renaming columns in R and just did it in Excel

missing_coords_split <- missing_coords |> split(missing_coords$Gene)

names(missing_coords_split) <- paste0(names(missing_coords_split),"_missing_coords")

missing_coords_split <- lapply(missing_coords_split, function(df) tidyr::pivot_longer(df, cols = c("Start", "Stop")))

#missing_coords_split <- lapply(missing_coords_split, function(df) rowid_to_column(df))

#missing_coords_split <- lapply(missing_coords_split, function(df) df[, colSums(is.na(df)) < nrow(df)])

#missing_coords_split <- lapply(missing_coords_split, function(df) rename(df, Boundary_Type = name))

missing_coords_split <- lapply(missing_coords_split, function(df) filter(df, is.na(value) == FALSE))

missing_coords_split <- lapply(missing_coords_split, function(df) tidyr::pivot_wider(df, names_from = name, values_from = value, values_fn = list))

missing_coords_split <- lapply(missing_coords_split, function(df) tidyr::unnest(df, cols = c("Start", "Stop")))

#missing_coords_split <- lapply(missing_coords_split, function(df) df[, colSums(is.na(df)) < nrow(df)])

list2env(missing_coords_split,envir=.GlobalEnv)

library(scales)

#unloadNamespace("flextable")

ama1plot <- ggplot(ama1_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(ama1_Tajima$BIN_START), max(ama1_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=AMA1_missing_coords$Start, xmax = AMA1_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvama1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("ama1_Tajima.png", ama1plot,device = png, width = 6, height = 3, units = 'in', res = 600)

cspplot <- ggplot(csp_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(csp_Tajima$BIN_START), max(csp_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=CSP_missing_coords$Start, xmax = CSP_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvcsp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("csp_Tajima.png", cspplot,device = png, width = 6, height = 3, units = 'in', res = 600)

dbpplot <- ggplot(dbp_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(dbp_Tajima$BIN_START), max(dbp_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=DBP_missing_coords$Start, xmax = DBP_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("dbp_Tajima.png", dbpplot,device = png, width = 6, height = 3, units = 'in', res = 600)

ebpplot <- ggplot(ebp_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(ebp_Tajima$BIN_START), max(ebp_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=EBP_missing_coords$Start, xmax = EBP_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvebp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("ebp_Tajima.png", ebpplot,device = png, width = 6, height = 3, units = 'in', res = 600)

gamaplot <- ggplot(gama_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(gama_Tajima$BIN_START), max(gama_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=GAMA_missing_coords$Start, xmax = GAMA_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvgama")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("gama_Tajima.png", gamaplot,device = png, width = 6, height = 3, units = 'in', res = 600)

msp1plot <- ggplot(msp1_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(msp1_Tajima$BIN_START), max(msp1_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=MSP1_missing_coords$Start, xmax = MSP1_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvmsp1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("msp1_Tajima.png", msp1plot,device = png, width = 6, height = 3, units = 'in', res = 600)

rbp1aplot <- ggplot(rbp1a_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(rbp1a_Tajima$BIN_START), max(rbp1a_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=RBP1a_missing_coords$Start, xmax = RBP1a_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("rbp1a_Tajima.png", rbp1aplot,device = png, width = 6, height = 3, units = 'in', res = 600)

rbp1bplot <- ggplot(rbp1b_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(rbp1b_Tajima$BIN_START), max(rbp1b_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
   annotate("rect", xmin=RBP1b_missing_coords$Start, xmax = RBP1b_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("rbp1b_Tajima.png", rbp1bplot,device = png, width = 6, height = 3, units = 'in', res = 600)

rbp2aplot <- ggplot(rbp2a_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(rbp2a_Tajima$BIN_START), max(rbp2a_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
   annotate("rect", xmin=RBP2a_missing_coords$Start, xmax = RBP2a_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("rbp2a_Tajima.png", rbp2aplot,device = png, width = 6, height = 3, units = 'in', res = 600)

rbp2bplot <- ggplot(rbp2b_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(rbp2b_Tajima$BIN_START), max(rbp2b_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
   annotate("rect", xmin=RBP2b_missing_coords$Start, xmax = RBP2b_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("rbp2b_Tajima.png", rbp2bplot,device = png, width = 6, height = 3, units = 'in', res = 600)

rbp2cplot <- ggplot(rbp2c_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(rbp2c_Tajima$BIN_START), max(rbp2c_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
   annotate("rect", xmin=RBP2c_missing_coords$Start, xmax = RBP2c_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2c")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("rbp2c_Tajima.png", rbp2cplot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag11plot <- ggplot(trag11_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3.25) +
  xlim(min(trag11_Tajima$BIN_START), max(trag11_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG11_missing_coords$Start, xmax = TRAG11_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag11")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag11_Tajima.png", trag11plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag19plot <- ggplot(trag19_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag19_Tajima$BIN_START), max(trag19_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG19_missing_coords$Start, xmax = TRAG19_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag19")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag19_Tajima.png", trag19plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag21plot <- ggplot(trag21_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag21_Tajima$BIN_START), max(trag21_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG21_missing_coords$Start, xmax = TRAG21_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag21")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag21_Tajima.png", trag21plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag22plot <- ggplot(trag22_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag22_Tajima$BIN_START), max(trag22_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG22_missing_coords$Start, xmax = TRAG22_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag22")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag22_Tajima.png", trag22plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag26plot <- ggplot(trag26_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag26_Tajima$BIN_START), max(trag26_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG26_missing_coords$Start, xmax = TRAG26_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag26")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag26_Tajima.png", trag26plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag32plot <- ggplot(trag32_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag32_Tajima$BIN_START), max(trag32_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG32_missing_coords$Start, xmax = TRAG32_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag32")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag32_Tajima.png", trag32plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag34plot <- ggplot(trag34_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag34_Tajima$BIN_START), max(trag34_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG34_missing_coords$Start, xmax = TRAG34_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag34")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag34_Tajima.png", trag34plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag36_2plot <- ggplot(trag36_2_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag36_2_Tajima$BIN_START), max(trag36_2_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=`TRAG36-2_missing_coords`$Start, xmax = `TRAG36-2_missing_coords`$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag36")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag36_Tajima.png", trag36_2plot,device = png, width = 6, height = 3, units = 'in', res = 600)

trag38plot <- ggplot(trag38_Tajima, aes(x = POS, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  xlim(min(trag38_Tajima$BIN_START), max(trag38_Tajima$BIN_END)) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  annotate("rect", xmin=TRAG38_missing_coords$Start, xmax = TRAG38_missing_coords$Stop, ymin = -3, ymax = 3, alpha = 0.2) +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag38")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size = 16), axis.text.y = element_text(size = 16), axis.title.y = element_text(size = 20), axis.title.x = element_text(size = 20), plot.title = element_text(size = 24))

ggsave("trag38_Tajima.png", trag38plot,device = png, width = 6, height = 3, units = 'in', res = 600)

#Tajima_plots <- paste0(unique(PvG_Tajima$gene), "plot")
#Tajima_plots <- Tajima_plots[!(Tajima_plots %in% c("crtplot", "dhfr-tsplot", "mdr1plot", "pppk-dhpsplot"))] |> sort()
#Tajima_plots <- replace(Tajima_plots, c(3,4), Tajima_plots[c(4,3)]) #need to do this to get dbp to show up before dbp2
#Tajima_plots <- list(mget(Tajima_plots))
#Tajima_plots <- list(ama1plot, cspplot, dbpplot, ebpplot, gamaplot, msp1plot, rbp1aplot, rbp1bplot, rbp2aplot, rbp2bplot, rbp2cplot, trag11plot, trag19plot, trag21plot, trag22plot, trag26plot, trag32plot, trag34plot, trag36_2plot, trag38plot)

#ggpubr::ggarrange(plotlist = Tajima_plots, labels = "AUTO", ncol = 4, nrow = 5, align = "hv")

#ggsave("Tajima_plots_all_for_supplement.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

library(patchwork)

design <- "
  ABCDE
  FGHIJ
  KLM##
"

ama1plot + dbpplot + ebpplot + msp1plot + rbp1aplot + rbp1bplot + rbp2aplot + rbp2bplot + rbp2cplot + trag11plot + trag19plot + trag22plot + trag36_2plot +
  plot_layout(design = design) +
  plot_annotation(tag_levels = "A")

ggsave("Tajima_balancing_selection.png", dpi = 600, width = 14, height = 10)

ama1plot + cspplot + dbpplot + ebpplot + gamaplot + msp1plot + rbp1aplot + rbp1bplot + rbp2aplot + rbp2bplot + rbp2cplot + trag11plot + trag19plot + trag21plot + trag22plot + trag26plot + trag32plot + trag34plot + trag36_2plot + trag38plot + plot_layout(ncol = 4, nrow = 5) + plot_annotation(tag_levels = "A")

ggsave("Tajima_plots_all_for_supplement.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

#Additional Tajima plots in chunks below

```

#Tajima with 10 bp window and 1 bp step
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
PvG_Tajima <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/PvG_vk_Tajima_10_sliding.txt", data.table = FALSE, header = TRUE) #|> subset(CHROM != "Transfer.PvP01_00_1.final")

PvG_Tajima <- PvG_Tajima |> dplyr::mutate(gene = dplyr::case_when(CHROM == "PvP01_09_v1" & BIN_START >= 1458201 & BIN_END <= 1460984 ~ "ama1",
                                                                  CHROM == "PvP01_01_v1" & BIN_START >= 440364 & BIN_END <= 446544 ~ "crt",
                                                               CHROM == "PvP01_08_v1" & BIN_START >= 1508714 & BIN_END <= 1509805 ~ "csp",
                                                               CHROM == "PvP01_06_v1" & BIN_START >= 981059 & BIN_END <= 987037 ~ "dbp",
                                                               CHROM == "PvP01_01_v1" & BIN_START >= 104013 & BIN_END <= 107429 ~ "dbp2",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 1075584 & BIN_END <= 1079966 ~ "dhfr-ts",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 252986 & BIN_END <= 256400 ~ "gama",
                                                               CHROM == "PvP01_10_v1" & BIN_START >= 476677 & BIN_END <= 483934 ~ "mdr1",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 1215432 & BIN_END <= 1220636 ~ "msp1",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 1269756 & BIN_END <= 1272657 ~ "pppk-dhps",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 71106 & BIN_END <= 80980 ~ "rbp1a",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 59612 & BIN_END <= 70478 ~ "rbp1b",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 112686 & BIN_END <= 121381 ~ "rbp2a",
                                                            CHROM == "PvP01_08_v1" & BIN_START >= 33312 & BIN_END <= 43704 ~ "rbp2b",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1458611 & BIN_END <= 1467388 ~ "rbp2c",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1390201 & BIN_END <= 1392765 ~ "trag11",
                                                            CHROM == "PvP01_11_v1" & BIN_START >= 68505 & BIN_END <= 71449 ~ "trag19",
                                                            CHROM == "PvP01_14_v1"& BIN_START >= 81068 & BIN_END <= 82841 ~ "trag21",
                                                            CHROM == "PvP01_14_v1" & BIN_START >= 2984025 & BIN_END <= 2988536 ~ "trag22",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 39887 & BIN_END <= 40931 ~ "trag26",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 72661 & BIN_END <= 75073 ~ "trag32",
                                                            CHROM == "PvP01_07_v1" & BIN_START >= 45445 & BIN_START >= 46410 ~ "trag34",
                                                            CHROM == "PvP01_09_v1" & BIN_START >= 2174904 & BIN_START >= 2176967 ~  "trag36_1",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 57969 & BIN_END <= 58929 ~ "trag36_2",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 150938 & BIN_END <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))

PvG_Tajima <- PvG_Tajima |> subset(gene != "not_targeted")

Tajima_split <- PvG_Tajima |> split(PvG_Tajima$gene)

names(Tajima_split) <- paste0(names(Tajima_split),"_Tajima")

list2env(Tajima_split,envir=.GlobalEnv)

library(scales)

#unloadNamespace("flextable")

ama1plot <- ggplot(ama1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvama1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

cspplot <- ggplot(csp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvcsp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbpplot <- ggplot(dbp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbp2plot <- ggplot(dbp2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp2")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

gamaplot <- ggplot(gama_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvgama")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

msp1plot <- ggplot(msp1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvmsp1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1aplot <- ggplot(rbp1a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1bplot <- ggplot(rbp1b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2aplot <- ggplot(rbp2a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2bplot <- ggplot(rbp2b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2cplot <- ggplot(rbp2c_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2c")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag11plot <- ggplot(trag11_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag11")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag19plot <- ggplot(trag19_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag19")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag21plot <- ggplot(trag21_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag21")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag22plot <- ggplot(trag22_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag22")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag26plot <- ggplot(trag26_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag26")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag32plot <- ggplot(trag32_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag32")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag34plot <- ggplot(trag34_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag34")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag36_2plot <- ggplot(trag36_2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag36")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag38plot <- ggplot(trag38_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag38")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

#Tajima_plots <- paste0(unique(PvG_Tajima$gene), "plot")
#Tajima_plots <- Tajima_plots[!(Tajima_plots %in% c("crtplot", "dhfr-tsplot", "mdr1plot", "pppk-dhpsplot"))] |> sort()
#Tajima_plots <- replace(Tajima_plots, c(3,4), Tajima_plots[c(4,3)]) #need to do this to get dbp to show up before dbp2
#Tajima_plots <- list(mget(Tajima_plots))
Tajima_plots <- list(ama1plot, cspplot, dbpplot, dbp2plot, gamaplot, msp1plot, rbp1aplot, rbp1bplot, rbp2aplot, rbp2bplot, rbp2cplot, trag11plot, trag19plot, trag21plot, trag22plot, trag26plot, trag32plot, trag34plot, trag36_2plot, trag38plot)

ggpubr::ggarrange(plotlist = Tajima_plots, labels = "AUTO", ncol = 4, nrow = 5, align = "hv")

ggsave("Tajima_plots_10_1.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

```


#Tajima with 10 bp window and 5 bp step
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
PvG_Tajima <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/PvG_vk_Tajima_10_5.txt", data.table = FALSE, header = TRUE) #|> subset(CHROM != "Transfer.PvP01_00_1.final")

PvG_Tajima <- PvG_Tajima |> dplyr::mutate(gene = dplyr::case_when(CHROM == "PvP01_09_v1" & BIN_START >= 1458201 & BIN_END <= 1460984 ~ "ama1",
                                                                  CHROM == "PvP01_01_v1" & BIN_START >= 440364 & BIN_END <= 446544 ~ "crt",
                                                               CHROM == "PvP01_08_v1" & BIN_START >= 1508714 & BIN_END <= 1509805 ~ "csp",
                                                               CHROM == "PvP01_06_v1" & BIN_START >= 981059 & BIN_END <= 987037 ~ "dbp",
                                                               CHROM == "PvP01_01_v1" & BIN_START >= 104013 & BIN_END <= 107429 ~ "dbp2",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 1075584 & BIN_END <= 1079966 ~ "dhfr-ts",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 252986 & BIN_END <= 256400 ~ "gama",
                                                               CHROM == "PvP01_10_v1" & BIN_START >= 476677 & BIN_END <= 483934 ~ "mdr1",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 1215432 & BIN_END <= 1220636 ~ "msp1",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 1269756 & BIN_END <= 1272657 ~ "pppk-dhps",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 71106 & BIN_END <= 80980 ~ "rbp1a",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 59612 & BIN_END <= 70478 ~ "rbp1b",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 112686 & BIN_END <= 121381 ~ "rbp2a",
                                                            CHROM == "PvP01_08_v1" & BIN_START >= 33312 & BIN_END <= 43704 ~ "rbp2b",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1458611 & BIN_END <= 1467388 ~ "rbp2c",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1390201 & BIN_END <= 1392765 ~ "trag11",
                                                            CHROM == "PvP01_11_v1" & BIN_START >= 68505 & BIN_END <= 71449 ~ "trag19",
                                                            CHROM == "PvP01_14_v1"& BIN_START >= 81068 & BIN_END <= 82841 ~ "trag21",
                                                            CHROM == "PvP01_14_v1" & BIN_START >= 2984025 & BIN_END <= 2988536 ~ "trag22",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 39887 & BIN_END <= 40931 ~ "trag26",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 72661 & BIN_END <= 75073 ~ "trag32",
                                                            CHROM == "PvP01_07_v1" & BIN_START >= 45445 & BIN_START >= 46410 ~ "trag34",
                                                            CHROM == "PvP01_09_v1" & BIN_START >= 2174904 & BIN_START >= 2176967 ~  "trag36_1",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 57969 & BIN_END <= 58929 ~ "trag36_2",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 150938 & BIN_END <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))

PvG_Tajima <- PvG_Tajima |> subset(gene != "not_targeted")

Tajima_split <- PvG_Tajima |> split(PvG_Tajima$gene)

names(Tajima_split) <- paste0(names(Tajima_split),"_Tajima")

list2env(Tajima_split,envir=.GlobalEnv)

library(scales)

#unloadNamespace("flextable")

ama1plot <- ggplot(ama1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvama1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

cspplot <- ggplot(csp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvcsp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbpplot <- ggplot(dbp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbp2plot <- ggplot(dbp2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp2")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

gamaplot <- ggplot(gama_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvgama")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

msp1plot <- ggplot(msp1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvmsp1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1aplot <- ggplot(rbp1a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1bplot <- ggplot(rbp1b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2aplot <- ggplot(rbp2a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2bplot <- ggplot(rbp2b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2cplot <- ggplot(rbp2c_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2c")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag11plot <- ggplot(trag11_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag11")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag19plot <- ggplot(trag19_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag19")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag21plot <- ggplot(trag21_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag21")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag22plot <- ggplot(trag22_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag22")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag26plot <- ggplot(trag26_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag26")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag32plot <- ggplot(trag32_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag32")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag34plot <- ggplot(trag34_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag34")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag36_2plot <- ggplot(trag36_2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag36")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag38plot <- ggplot(trag38_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag38")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

#Tajima_plots <- paste0(unique(PvG_Tajima$gene), "plot")
#Tajima_plots <- Tajima_plots[!(Tajima_plots %in% c("crtplot", "dhfr-tsplot", "mdr1plot", "pppk-dhpsplot"))] |> sort()
#Tajima_plots <- replace(Tajima_plots, c(3,4), Tajima_plots[c(4,3)]) #need to do this to get dbp to show up before dbp2
#Tajima_plots <- list(mget(Tajima_plots))
Tajima_plots <- list(ama1plot, cspplot, dbpplot, dbp2plot, gamaplot, msp1plot, rbp1aplot, rbp1bplot, rbp2aplot, rbp2bplot, rbp2cplot, trag11plot, trag19plot, trag21plot, trag22plot, trag26plot, trag32plot, trag34plot, trag36_2plot, trag38plot)

ggpubr::ggarrange(plotlist = Tajima_plots, labels = "AUTO", ncol = 4, nrow = 5, align = "hv")

ggsave("Tajima_plots_10_5.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

```


#Tajima with 50 bp window and 25 bp step
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
PvG_Tajima <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/PvG_vk_Tajima_50_25.txt", data.table = FALSE, header = TRUE) #|> subset(CHROM != "Transfer.PvP01_00_1.final")

PvG_Tajima <- PvG_Tajima |> dplyr::mutate(gene = dplyr::case_when(CHROM == "PvP01_09_v1" & BIN_START >= 1458201 & BIN_END <= 1460984 ~ "ama1",
                                                                  CHROM == "PvP01_01_v1" & BIN_START >= 440364 & BIN_END <= 446544 ~ "crt",
                                                               CHROM == "PvP01_08_v1" & BIN_START >= 1508714 & BIN_END <= 1509805 ~ "csp",
                                                               CHROM == "PvP01_06_v1" & BIN_START >= 981059 & BIN_END <= 987037 ~ "dbp",
                                                               CHROM == "PvP01_01_v1" & BIN_START >= 104013 & BIN_END <= 107429 ~ "dbp2",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 1075584 & BIN_END <= 1079966 ~ "dhfr-ts",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 252986 & BIN_END <= 256400 ~ "gama",
                                                               CHROM == "PvP01_10_v1" & BIN_START >= 476677 & BIN_END <= 483934 ~ "mdr1",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 1215432 & BIN_END <= 1220636 ~ "msp1",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 1269756 & BIN_END <= 1272657 ~ "pppk-dhps",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 71106 & BIN_END <= 80980 ~ "rbp1a",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 59612 & BIN_END <= 70478 ~ "rbp1b",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 112686 & BIN_END <= 121381 ~ "rbp2a",
                                                            CHROM == "PvP01_08_v1" & BIN_START >= 33312 & BIN_END <= 43704 ~ "rbp2b",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1458611 & BIN_END <= 1467388 ~ "rbp2c",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1390201 & BIN_END <= 1392765 ~ "trag11",
                                                            CHROM == "PvP01_11_v1" & BIN_START >= 68505 & BIN_END <= 71449 ~ "trag19",
                                                            CHROM == "PvP01_14_v1"& BIN_START >= 81068 & BIN_END <= 82841 ~ "trag21",
                                                            CHROM == "PvP01_14_v1" & BIN_START >= 2984025 & BIN_END <= 2988536 ~ "trag22",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 39887 & BIN_END <= 40931 ~ "trag26",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 72661 & BIN_END <= 75073 ~ "trag32",
                                                            CHROM == "PvP01_07_v1" & BIN_START >= 45445 & BIN_START >= 46410 ~ "trag34",
                                                            CHROM == "PvP01_09_v1" & BIN_START >= 2174904 & BIN_START >= 2176967 ~  "trag36_1",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 57969 & BIN_END <= 58929 ~ "trag36_2",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 150938 & BIN_END <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))

PvG_Tajima <- PvG_Tajima |> subset(gene != "not_targeted")

Tajima_split <- PvG_Tajima |> split(PvG_Tajima$gene)

names(Tajima_split) <- paste0(names(Tajima_split),"_Tajima")

list2env(Tajima_split,envir=.GlobalEnv)

library(scales)

#unloadNamespace("flextable")

ama1plot <- ggplot(ama1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvama1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

cspplot <- ggplot(csp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvcsp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbpplot <- ggplot(dbp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbp2plot <- ggplot(dbp2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp2")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

gamaplot <- ggplot(gama_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvgama")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

msp1plot <- ggplot(msp1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvmsp1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1aplot <- ggplot(rbp1a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1bplot <- ggplot(rbp1b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2aplot <- ggplot(rbp2a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2bplot <- ggplot(rbp2b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2cplot <- ggplot(rbp2c_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2c")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag11plot <- ggplot(trag11_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag11")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag19plot <- ggplot(trag19_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag19")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag21plot <- ggplot(trag21_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag21")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag22plot <- ggplot(trag22_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag22")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag26plot <- ggplot(trag26_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag26")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag32plot <- ggplot(trag32_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag32")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag34plot <- ggplot(trag34_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag34")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag36_2plot <- ggplot(trag36_2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag36")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag38plot <- ggplot(trag38_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag38")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

#Tajima_plots <- paste0(unique(PvG_Tajima$gene), "plot")
#Tajima_plots <- Tajima_plots[!(Tajima_plots %in% c("crtplot", "dhfr-tsplot", "mdr1plot", "pppk-dhpsplot"))] |> sort()
#Tajima_plots <- replace(Tajima_plots, c(3,4), Tajima_plots[c(4,3)]) #need to do this to get dbp to show up before dbp2
#Tajima_plots <- list(mget(Tajima_plots))
Tajima_plots <- list(ama1plot, cspplot, dbpplot, dbp2plot, gamaplot, msp1plot, rbp1aplot, rbp1bplot, rbp2aplot, rbp2bplot, rbp2cplot, trag11plot, trag19plot, trag21plot, trag22plot, trag26plot, trag32plot, trag34plot, trag36_2plot, trag38plot)

ggpubr::ggarrange(plotlist = Tajima_plots, labels = "AUTO", ncol = 4, nrow = 5, align = "hv")

ggsave("Tajima_plots_50_25.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

```

#Tajima with 100 bp window and 50 bp step
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")
PvG_Tajima <- data.table::fread("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/PvG_vk_Tajima_100_50.txt", data.table = FALSE, header = TRUE) #|> subset(CHROM != "Transfer.PvP01_00_1.final")

PvG_Tajima <- PvG_Tajima |> dplyr::mutate(gene = dplyr::case_when(CHROM == "PvP01_09_v1" & BIN_START >= 1458201 & BIN_END <= 1460984 ~ "ama1",
                                                                  CHROM == "PvP01_01_v1" & BIN_START >= 440364 & BIN_END <= 446544 ~ "crt",
                                                               CHROM == "PvP01_08_v1" & BIN_START >= 1508714 & BIN_END <= 1509805 ~ "csp",
                                                               CHROM == "PvP01_06_v1" & BIN_START >= 981059 & BIN_END <= 987037 ~ "dbp",
                                                               CHROM == "PvP01_01_v1" & BIN_START >= 104013 & BIN_END <= 107429 ~ "dbp2",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 1075584 & BIN_END <= 1079966 ~ "dhfr-ts",
                                                               CHROM == "PvP01_05_v1" & BIN_START >= 252986 & BIN_END <= 256400 ~ "gama",
                                                               CHROM == "PvP01_10_v1" & BIN_START >= 476677 & BIN_END <= 483934 ~ "mdr1",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 1215432 & BIN_END <= 1220636 ~ "msp1",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 1269756 & BIN_END <= 1272657 ~ "pppk-dhps",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 71106 & BIN_END <= 80980 ~ "rbp1a",
                                                               CHROM == "PvP01_07_v1" & BIN_START >= 59612 & BIN_END <= 70478 ~ "rbp1b",
                                                               CHROM == "PvP01_14_v1" & BIN_START >= 112686 & BIN_END <= 121381 ~ "rbp2a",
                                                            CHROM == "PvP01_08_v1" & BIN_START >= 33312 & BIN_END <= 43704 ~ "rbp2b",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1458611 & BIN_END <= 1467388 ~ "rbp2c",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 1390201 & BIN_END <= 1392765 ~ "trag11",
                                                            CHROM == "PvP01_11_v1" & BIN_START >= 68505 & BIN_END <= 71449 ~ "trag19",
                                                            CHROM == "PvP01_14_v1"& BIN_START >= 81068 & BIN_END <= 82841 ~ "trag21",
                                                            CHROM == "PvP01_14_v1" & BIN_START >= 2984025 & BIN_END <= 2988536 ~ "trag22",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 39887 & BIN_END <= 40931 ~ "trag26",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 72661 & BIN_END <= 75073 ~ "trag32",
                                                            CHROM == "PvP01_07_v1" & BIN_START >= 45445 & BIN_START >= 46410 ~ "trag34",
                                                            CHROM == "PvP01_09_v1" & BIN_START >= 2174904 & BIN_START >= 2176967 ~  "trag36_1",
                                                            CHROM == "Transfer.PvP01_00_1.final" & BIN_START >= 57969 & BIN_END <= 58929 ~ "trag36_2",
                                                            CHROM == "PvP01_05_v1" & BIN_START >= 150938 & BIN_END <= 152029 ~ "trag38",
                                                            .default = "not_targeted"))

PvG_Tajima <- PvG_Tajima |> subset(gene != "not_targeted")

PvG_Tajima |> data.table::fwrite("Tajima_100_50.txt")

Tajima_split <- PvG_Tajima |> split(PvG_Tajima$gene)

names(Tajima_split) <- paste0(names(Tajima_split),"_Tajima")

list2env(Tajima_split,envir=.GlobalEnv)

library(scales)

#unloadNamespace("flextable")

ama1plot <- ggplot(ama1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvama1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

cspplot <- ggplot(csp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvcsp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbpplot <- ggplot(dbp_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

dbp2plot <- ggplot(dbp2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvdbp2")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

gamaplot <- ggplot(gama_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvgama")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

msp1plot <- ggplot(msp1_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvmsp1")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1aplot <- ggplot(rbp1a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp1bplot <- ggplot(rbp1b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp1b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2aplot <- ggplot(rbp2a_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2a")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2bplot <- ggplot(rbp2b_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2b")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

rbp2cplot <- ggplot(rbp2c_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvrbp2c")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag11plot <- ggplot(trag11_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag11")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag19plot <- ggplot(trag19_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag19")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag21plot <- ggplot(trag21_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag21")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag22plot <- ggplot(trag22_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag22")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag26plot <- ggplot(trag26_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag26")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag32plot <- ggplot(trag32_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag32")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag34plot <- ggplot(trag34_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag34")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag36_2plot <- ggplot(trag36_2_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag36")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

trag38plot <- ggplot(trag38_Tajima, aes(x = BIN_START, y = TajimaD)) + 
  geom_point(aes(color = TajimaD > abs(2)), show.legend = FALSE) +
  scale_color_manual(values = c("black", "red")) +
  theme_classic() +
  ylim(-3,3) +
  labs(y= "Tajima's D", x = "Position") +
  geom_hline(yintercept = 0) +
  geom_hline(yintercept = 2, linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -2, linetype = "dashed", color = "blue") +
  scale_x_continuous(label = comma) +
  ggtitle(expression(paste(italic("Pvtrag38")))) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

#Tajima_plots <- paste0(unique(PvG_Tajima$gene), "plot")
#Tajima_plots <- Tajima_plots[!(Tajima_plots %in% c("crtplot", "dhfr-tsplot", "mdr1plot", "pppk-dhpsplot"))] |> sort()
#Tajima_plots <- replace(Tajima_plots, c(3,4), Tajima_plots[c(4,3)]) #need to do this to get dbp to show up before dbp2
#Tajima_plots <- list(mget(Tajima_plots))
Tajima_plots <- list(ama1plot, cspplot, dbpplot, dbp2plot, gamaplot, msp1plot, rbp1aplot, rbp1bplot, rbp2aplot, rbp2bplot, rbp2cplot, trag11plot, trag19plot, trag21plot, trag22plot, trag26plot, trag32plot, trag34plot, trag36_2plot, trag38plot)

ggpubr::ggarrange(plotlist = Tajima_plots, labels = "AUTO", ncol = 4, nrow = 5, align = "hv")

ggsave("Tajima_plots_100_50.png", device = "png", dpi = 600, width = 30, height = 20, units = "in")

```


#MIP coverage heatmap for tiled genes
```{r}
setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/PvG_wrangler/analysis/")

library(dplyr)
library(ggpubr)
library(tidyverse)
library(ggplot2)
library(viridis)

barcode_counts <- data.table::fread("barcode_counts.csv", data.table = F)

barcode_counts <- barcode_counts[-c(1,2),] #remove first two rows which don't contain data or useful labels

barcode_counts <- barcode_counts |> dplyr::rename("Sample" = "MIP") |> dplyr::filter(!grepl("SNP", Sample))

barcode_counts$Sample <- barcode_counts$Sample |> stringr::str_remove(pattern = "-PVG")

barcode_counts <- barcode_counts |> dplyr::filter(!grepl("NTC", Sample)) |> dplyr::filter(!grepl("NTP", Sample))

barcode_counts <- barcode_counts |> column_to_rownames(var = "Sample")

barcode_counts <- barcode_counts[gtools::mixedorder(colnames(barcode_counts))] #puts MIPs in proper numerical order rather than going from 1 to 10 and skipping 2-9

colnames(barcode_counts) <- colnames(barcode_counts) |> stringr::str_remove(pattern = "_S0_Sub0")

barcode_tibble <- barcode_counts |> as.tibble(rownames = NA) 

barcode_tibble <- barcode_tibble |> rownames_to_column(var = "Sample")

barcode_tibble <- barcode_tibble |> pivot_longer(!Sample, names_to = "mip", values_to = "count")

barcode_tibble$count <- as.numeric(barcode_tibble$count)

#correct_order <- factor(barcode_tibble$mip, levels = unique(barcode_tibble$mip))

barcode_tibble |> ggplot(aes(mip, Sample, fill = count)) + geom_tile() #this is uninterpretable, need to split by gene

barcode_tibble <- barcode_tibble |> dplyr::mutate(Gene = stringr::str_extract(barcode_tibble$mip, pattern = "[^_]+"))

barcode_tibble$mip <- factor(barcode_tibble$mip, levels = unique(barcode_tibble$mip))

barcode_tibble <- barcode_tibble |> dplyr::mutate(logcount = log(count))

barcode_split <- barcode_tibble |> split(barcode_tibble$Gene)

names(barcode_split) <- paste0(names(barcode_split),"_barcode")

list2env(barcode_split,envir=.GlobalEnv)

setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Pvivax MIP Design/Analysis/NAMRU_repool/Combined_Analysis/")

AMA1_heatmap <- AMA1_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

#ggsave("AMA1_heatmap.png", AMA1_heatmap, width = 20, height = 10, dpi = 600, units = "mm")

CRT_heatmap <- CRT_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

CSP_heatmap <- CSP_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

DBP_heatmap <- DBP_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

DBP2_heatmap <- DBP2_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

DHFR_heatmap <- `DHFR-TS_barcode` |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

GAMA_heatmap <- GAMA_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

MDR1_heatmap <- MDR1_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

MSP1_heatmap <- MSP1_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

DHPS_heatmap <- `PPPK-DHPS_barcode` |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

RBP1a_heatmap <- RBP1a_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

RBP1b_heatmap <- RBP1b_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

RBP2a_heatmap <- RBP2a_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

RBP2b_heatmap <- RBP2b_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

RBP2c_heatmap <- RBP2c_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG11_heatmap <- TRAG11_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG19_heatmap <- TRAG19_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG21_heatmap <- TRAG21_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG22_heatmap <- TRAG22_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG26_heatmap <- TRAG26_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG32_heatmap <- TRAG32_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG36_1_heatmap <- `TRAG36-1_barcode` |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG36_2_heatmap <- `TRAG36-2_barcode` |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

TRAG38_heatmap <- TRAG38_barcode |> ggplot(aes(mip, Sample, fill = logcount)) + geom_tile() + theme_void() + theme(axis.text.x = element_text(angle = 90, size = 16)) + theme(plot.margin = unit(c(2,2,2,2), "mm"), legend.position = "none")

all_heatmaps <- list(AMA1_heatmap, CRT_heatmap, CSP_heatmap, DBP_heatmap, DBP2_heatmap, DHFR_heatmap, DHPS_heatmap, GAMA_heatmap, MDR1_heatmap, MSP1_heatmap, RBP1a_heatmap, RBP1b_heatmap, RBP2a_heatmap, RBP2b_heatmap, RBP2c_heatmap, TRAG11_heatmap, TRAG19_heatmap, TRAG21_heatmap, TRAG22_heatmap, TRAG26_heatmap, TRAG32_heatmap, TRAG36_1_heatmap, TRAG36_2_heatmap, TRAG38_heatmap)

#ggpubr::ggarrange(plotlist = all_heatmaps, labels = "AUTO", ncol = 5, nrow = 5, align = "hv", common.legend = T, legend = "bottom", vjust = -0.1, hjust = -0.4) + theme(plot.margin = unit(c(1,1,1,1), "in"), plot.background = element_rect(fill = "white", color = "white", linewidth = 0))

library(patchwork)

design <- "
AABBBCDDD#
EEFFFFGGHH
IIIIIIJJJ#
KKKKKMMMM#
LLLLLLLOOO
NNNNNPPPQQ
RRSSSTUUVV
WXYYYYYYYY
"

AMA1_heatmap + CRT_heatmap + CSP_heatmap + DBP_heatmap + DBP2_heatmap + DHFR_heatmap + DHPS_heatmap + GAMA_heatmap + MDR1_heatmap + MSP1_heatmap + RBP1a_heatmap + RBP1b_heatmap + RBP2a_heatmap + RBP2b_heatmap + RBP2c_heatmap + TRAG11_heatmap + TRAG19_heatmap + TRAG21_heatmap + TRAG22_heatmap + TRAG26_heatmap + TRAG32_heatmap + TRAG36_1_heatmap + TRAG36_2_heatmap + TRAG38_heatmap + guide_area() + plot_layout(design = design, guides = "collect") + plot_annotation(tag_levels = "A") & scale_fill_viridis(discrete = F, limits = c(0, max(barcode_tibble$logcount))) & theme(legend.position = "bottom", legend.key.size = unit(2, "in"), legend.title = element_text(size = 36), legend.text = element_text(size = 20), plot.tag = element_text(size = 20))

ggsave("all_heatmaps.svg", dpi = 600, width = 30, height = 25, units = "in")

ggsave("all_heatmaps.png", dpi = 600, width = 30, height = 25, units = "in")

#saving as SVG so I can manually move the legend where I want it, then saving that version as a PNG in Inkscape

```

```{r}
library(tidyverse)
library(igraph)
library(showtext)
library(rcartocolor)

setwd("C:/Users/zpopkinh/OneDrive - University of North Carolina at Chapel Hill/Manuscripts/vivax MIPs/")

sample_flow <- tibble(from = c("Initial Capture", "Initial Capture", "Successful SNP Captures", "Successful SNP Captures", "Successful SNP Captures"),
                      to = c("Successful PvG Captures", "Successful SNP Captures", "COI Dataset", "PCA Dataset", "CNV Dataset"))

sample_graph <- graph_from_data_frame(sample_flow, directed = TRUE)
coords <- layout_as_tree(sample_graph)
colnames(coords) <- c("x", "y")

output_df  <-  as_tibble(coords) |>
  mutate(step = vertex_attr(sample_graph, "name"),
         label = c("866 Initial Captures", "755 Successful SNP Captures", "740 Successful PvG Captures", "697 Samples for COI", "685 Samples for PCA", "668 Samples for CNV"),
         x = x*-1,
         type = factor(c(1, 2, 2, 3, 3, 3)))

plot_nodes  <-  output_df |>
  mutate(xmin = x - 0.25,
         xmax = x + 0.25,
         ymin = y - 0.05,
         ymax = y + 0.05)

plot_edges  <-  sample_flow |>
  mutate(id = row_number()) |>
  pivot_longer(cols = c("from", "to"),
               names_to = "s_e",
               values_to = "step")|>
  left_join(plot_nodes, by = "step") |>
  select(-c(label, type, y, xmin, xmax))|>
  mutate(y = ifelse(s_e == "from", ymin, ymax)) |>
  select(-c(ymin, ymax))

flowchart <-  ggplot() +
  theme_void() +
  geom_rect(data = plot_nodes,
            mapping = aes(xmin = xmin, ymin = ymin, 
                          xmax = xmax, ymax = ymax, 
                          fill = type, colour = type),
             alpha = 0.75) +
  geom_text(data = plot_nodes,
            mapping = aes(x = x, y = y, label = label, size = 36)) +
  geom_path(data = plot_edges,
            mapping = aes(x = x, y = y, group = id),
            arrow = arrow(length = unit(0.3, "cm"), type = "closed")) +
  scale_fill_carto_d(palette= "Earth") +
  scale_color_carto_d(palette = "Earth") +
  theme(legend.position = "none", plot.background = element_rect(fill = "white", color = "black"))

ggsave("FigS3_Flowchart.png", dpi = 600)

```

